<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Sun Jul 06 18:05:53 EDT 2014 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
Index (sequtils 0.92-SNAPSHOT API)
</TITLE>

<META NAME="date" CONTENT="2014-07-06">

<LINK REL ="stylesheet" TYPE="text/css" HREF="./stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Index (sequtils 0.92-SNAPSHOT API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="./index.html?index-all.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-all.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="./allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="./allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<A HREF="#_A_">A</A> <A HREF="#_B_">B</A> <A HREF="#_C_">C</A> <A HREF="#_D_">D</A> <A HREF="#_E_">E</A> <A HREF="#_F_">F</A> <A HREF="#_G_">G</A> <A HREF="#_H_">H</A> <A HREF="#_I_">I</A> <A HREF="#_J_">J</A> <A HREF="#_K_">K</A> <A HREF="#_L_">L</A> <A HREF="#_M_">M</A> <A HREF="#_N_">N</A> <A HREF="#_O_">O</A> <A HREF="#_P_">P</A> <A HREF="#_R_">R</A> <A HREF="#_S_">S</A> <A HREF="#_T_">T</A> <A HREF="#_U_">U</A> <A HREF="#_V_">V</A> <HR>
<A NAME="_A_"><!-- --></A><H2>
<B>A</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>AbstractSequenceFragmentIterator</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides an interface for an Iterator of Kcounts<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#AbstractSequenceFragmentIterator()"><B>AbstractSequenceFragmentIterator()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#add(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>add(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#add(byte[])"><B>add(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Gets the child node associated with the given sequence, creating it (and nodes along the way) as needed
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addAll(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet)"><B>addAll(FastaFileSet)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#addChild(byte)"><B>addChild(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Gets the child node associated with the given symbol, creating it first if needed.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#addChild(byte)"><B>addChild(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>Gets the child node associated with the given symbol, creating it first if needed.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addDirectory(java.io.File)"><B>addDirectory(File)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addFile(java.io.File)"><B>addFile(File)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#addPseudocounts()"><B>addPseudocounts()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#addUnknown()"><B>addUnknown()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Adds an "unknown" sample to this kcount, indicating that a character was consumed without incrementing any counter.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#addUpstreamNode(byte[])"><B>addUpstreamNode(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets the upstream node associated with the given suffix starting from this node, creating it (and nodes along the
 way) as needed
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#addUpstreamNode(byte)"><B>addUpstreamNode(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets the upstream node associated with the given symbol starting from this node, creating it first if needed.
<DT><A HREF="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html" title="interface in edu.berkeley.compbio.sequtils"><B>AffineSubstitutionMatrix</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>AggregateSequenceFragment</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#AggregateSequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, java.util.Collection)"><B>AggregateSequenceFragment(SequenceFragmentMetadata, String, Collection&lt;SequenceFragment&gt;)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>Constructs a new SequenceFragment by specifying its coordinates with respect to a containing parent sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#AggregateSequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>AggregateSequenceFragment(SequenceFragmentMetadata, String, SequenceFragment)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#align(byte[], byte[])"><B>align(byte[], byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#alphabet"><B>alphabet</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#alphabet"><B>alphabet</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#appendString(java.util.Formatter, java.lang.String)"><B>appendString(Formatter, String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#appendString(java.util.Formatter, java.lang.String)"><B>appendString(Formatter, String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#appendString(java.util.Formatter, java.lang.String)"><B>appendString(Formatter, String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Appends a string representation of this node to the given formatter, indenting as needed based on the hierarchy
 level.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#applyMask(boolean[], byte[], byte)"><B>applyMask(boolean[], byte[], byte)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>copy a slice ending at the given position (exclusive!) of whatever width is needed to obtain the requested number of
 non-gap characters.
</DL>
<HR>
<A NAME="_B_"><!-- --></A><H2>
<B>B</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#buf"><B>buf</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#bufPosition"><B>bufPosition</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><B>ByteArraySequenceReader</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#ByteArraySequenceReader(byte[])"><B>ByteArraySequenceReader(byte[])</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#ByteArraySequenceReader(java.lang.String)"><B>ByteArraySequenceReader(String)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_C_"><!-- --></A><H2>
<B>C</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#checkAvailable()"><B>checkAvailable()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#children"><B>children</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>Clones this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>Clone this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>Clone this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Performs a deep copy of this node.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Clone this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#clone()"><B>clone()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Clone this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#clone()"><B>clone()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Closes the reader, and all sub-readers
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Closes the reader, and all sub-readers
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#close()"><B>close()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>Closes the SequenceFragmentIterator and all sub-readers
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#close()"><B>close()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Closes the reader, and all sub-readers
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#close()"><B>close()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#columnToSeqIndex(byte[], int)"><B>columnToSeqIndex(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#compareTo(C)"><B>compareTo(C)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Compares this object with the specified object for order.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#completionProcessor"><B>completionProcessor</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte)"><B>conditionalProbability(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte[], byte[])"><B>conditionalProbability(byte[], byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte, byte[])"><B>conditionalProbability(byte, byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#conditionalProbability(byte, byte[])"><B>conditionalProbability(byte, byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Computes the conditional probability of generating a symbol given a prefix under the model, backing off to shorter
 prefixes as needed if the given prefix is not explicitly represented.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#conditionalProbability(byte, byte[])"><B>conditionalProbability(byte, byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Computes the conditional probability of generating a symbol given a prefix under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbabilityByAlphabetIndex(int)"><B>conditionalProbabilityByAlphabetIndex(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalsFrom(byte[])"><B>conditionalsFrom(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Computes the conditional probability distribution of symbols given a prefix under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#conditionalsFrom(byte[])"><B>conditionalsFrom(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Computes the conditional probability distribution of symbols given a prefix under the model, backing off to shorter
 prefixes as needed if the given prefix is not explicitly represented.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#conditionalsFrom(byte[])"><B>conditionalsFrom(byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Computes the conditional probability distribution of symbols given a prefix under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copyNoGaps(byte[])"><B>copyNoGaps(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#copyProbsFrom(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>copyProbsFrom(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Recursively fills out the tree so that each node has either a complete set of probabilities or none at all, and
 assigns conditional probabilities according to the given spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#copyProbsFromSpectrumRecursively(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>copyProbsFromSpectrumRecursively(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Recursively fills out the tree so that each node has either a complete set of children or none at all, and assigns
 conditional probabilities according to the given spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySlice(byte[], int, int)"><B>copySlice(byte[], int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySliceUpToNNonGaps(byte[], int, int)"><B>copySliceUpToNNonGaps(byte[], int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>copy a slice starting at the given position of whatever width is needed to obtain the requested number of non-gap
 characters
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copyWithout(byte[], byte)"><B>copyWithout(byte[], byte)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#countChildren()"><B>countChildren()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#counts"><B>counts</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#counts"><B>counts</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#countUpstreamNodes()"><B>countUpstreamNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Counts the immediately upstream nodes that exist.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#currentSection"><B>currentSection</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#currentSectionPosition"><B>currentSectionPosition</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_D_"><!-- --></A><H2>
<B>D</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#dataSum"><B>dataSum</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#decrementBy(T)"><B>decrementBy(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>updates this object by subtracting another one from it.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#decrementBy(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)"><B>decrementBy(MarkovTreeNode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#decrementBy(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>decrementBy(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#decrementBy(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>decrementBy(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode, double)"><B>decrementByWeighted(MarkovTreeNode, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.RonPST, double)"><B>decrementByWeighted(RonPST, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.SequenceFragment, double)"><B>decrementByWeighted(SequenceFragment, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#desiredLengthUnknown()"><B>desiredLengthUnknown()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#diagnostics()"><B>diagnostics()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html#distanceFromTo(byte[], byte[])"><B>distanceFromTo(byte[], byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/GapFrequency.html#distanceFromTo(byte[], byte[])"><B>distanceFromTo(byte[], byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils">GapFrequency</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#distanceFromTo(byte[], byte[])"><B>distanceFromTo(byte[], byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils"><B>DnaDist</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html#DnaDist(double, double, int)"><B>DnaDist(double, double, int)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#doneLabelling()"><B>doneLabelling()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#doneLabelling()"><B>doneLabelling()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>DoubleKcount</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="type parameter in DoubleKcount">T</A> extends <A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Represents the counts of a set of given patterns appearing in a string.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#DoubleKcount(double[])"><B>DoubleKcount(double[])</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>Creates a new Kcount instance.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>DoublePseudocountAdder</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#DoublePseudocountAdder()"><B>DoublePseudocountAdder()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils"><B>DynamicProgrammingPairwiseAligner</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#DynamicProgrammingPairwiseAligner(edu.berkeley.compbio.sequtils.AffineSubstitutionMatrix, edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner.TracebackBegin, edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner.TracebackEnd)"><B>DynamicProgrammingPairwiseAligner(AffineSubstitutionMatrix, DynamicProgrammingPairwiseAligner.TracebackBegin, DynamicProgrammingPairwiseAligner.TracebackEnd)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils"><B>DynamicProgrammingPairwiseAligner.TracebackBegin</B></A> - Enum in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils"><B>DynamicProgrammingPairwiseAligner.TracebackEnd</B></A> - Enum in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;</DL>
<HR>
<A NAME="_E_"><!-- --></A><H2>
<B>E</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/package-summary.html"><B>edu.berkeley.compbio.sequtils</B></A> - package edu.berkeley.compbio.sequtils<DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html"><B>edu.berkeley.compbio.sequtils.sequencefragmentiterator</B></A> - package edu.berkeley.compbio.sequtils.sequencefragmentiterator<DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html"><B>edu.berkeley.compbio.sequtils.sequencereader</B></A> - package edu.berkeley.compbio.sequtils.sequencereader<DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html"><B>edu.berkeley.compbio.sequtils.sequencereader.fasta</B></A> - package edu.berkeley.compbio.sequtils.sequencereader.fasta<DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html"><B>edu.berkeley.compbio.sequtils.strings</B></A> - package edu.berkeley.compbio.sequtils.strings<DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#endsWithGaps(byte[], int)"><B>endsWithGaps(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#ensureAllParentsExist()"><B>ensureAllParentsExist()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>Recursively generalize thisKcount, creating a chain of "parents" until no further generalization is possible
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#equalValue(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>equalValue(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#equalValue(T)"><B>equalValue(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>Tests whether the given pattern counts are equal to those stored here.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#equalValue(T)"><B>equalValue(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>Test whether the given pattern counts are equal to those stored here.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#equalValue(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)"><B>equalValue(MarkovTreeNode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Recursively tests the deep equality of two PSTNodes.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#equalValue(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>equalValue(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#equalValue(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>equalValue(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Test whether the given object is the same as this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</A>
<DD>Note this is partly redundant with RepeatingAggregatingSFI
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>provides the next item from the iterator without actually consuming it
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</A>
<DD>Note this is partly redundant with RepeatingAggregatingSFI
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#estimatedTotalSamples()"><B>estimatedTotalSamples()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_F_"><!-- --></A><H2>
<B>F</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>FastaFileSet</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</A><DD>Fasta File Set, useful in treating multiple fasta files as one<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.lang.Integer)"><B>FastaFileSet(Integer)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.lang.String)"><B>FastaFileSet(String)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.io.File)"><B>FastaFileSet(File)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Initializes a file set from a file or directory
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.util.Collection)"><B>FastaFileSet(Collection&lt;File&gt;)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Initializes a file set from a file or directory
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.io.File, java.lang.Integer)"><B>FastaFileSet(File, Integer)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.util.Collection, java.lang.Integer)"><B>FastaFileSet(Collection&lt;FastaFileSet&gt;, Integer)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD><B>Deprecated.</B>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>FastaParser</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</A><DD>Fasta File Parser<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, boolean)"><B>FastaParser(FastaFileSet, boolean)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Initializes a parser from a fileset
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, int, boolean)"><B>FastaParser(FastaFileSet, int, boolean)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Initializes a parser from a fileset and with a specific buffer size
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>FastaParserSet</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</A><DD>Creates a HashSet of FastaParsers<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.util.Collection)"><B>FastaParserSet(Collection&lt;File&gt;)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.io.File)"><B>FastaParserSet(File)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.io.File, int)"><B>FastaParserSet(File, int)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#filename"><B>filename</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte)"><B>filter(byte)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</A>
<DD>Mutate the provided nucleotide and return it.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte[])"><B>filter(byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</A>
<DD>Mutate each nucleotide in the provided byte buffer, in place.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte[], int)"><B>filter(byte[], int)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</A>
<DD>Mutate the first n nucleotides in the provided byte buffer, in place.
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils"><B>FilterException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.String)"><B>FilterException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.Exception)"><B>FilterException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.Exception, java.lang.String)"><B>FilterException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader"><B>FilteringSectionList</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#FilteringSectionList(edu.berkeley.compbio.sequtils.sequencereader.SectionList, edu.berkeley.compbio.sequtils.SubstitutionFilter)"><B>FilteringSectionList(SectionList, SubstitutionFilter)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><B>FilteringSequenceReader</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#FilteringSequenceReader(edu.berkeley.compbio.sequtils.SequenceReader, edu.berkeley.compbio.sequtils.SubstitutionFilter)"><B>FilteringSequenceReader(SequenceReader, SubstitutionFilter)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#fireUpdated(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>fireUpdated(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Indicates that the sequence statistics have changed in some way, such that any derived statistics must be
 recomputed.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html" title="interface in edu.berkeley.compbio.sequtils.strings"><B>FirstWordProvider</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Interface for objects that are able to provide a list of the "first words" associated with a sequence spectrum.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment, boolean)"><B>fragmentLogProbability(SequenceFragment, boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Computes the total log probability of generating the given sequence fragment under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment, boolean)"><B>fragmentLogProbability(SequenceFragment, boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</A>
<DD>Computes the total log probability of generating the given sequence fragment under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment, boolean)"><B>fragmentLogProbability(SequenceFragment, boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Computes the total log probability of generating the given sequence fragment under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment, boolean)"><B>fragmentLogProbability(SequenceFragment, boolean)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Computes the total log probability of generating the given sequence fragment under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#fragmentLogProbabilityFromIntKcount(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>fragmentLogProbabilityFromIntKcount(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Computes the total log probability of generating the given sequence fragment under the model.
</DL>
<HR>
<A NAME="_G_"><!-- --></A><H2>
<B>G</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#GAP_BYTE"><B>GAP_BYTE</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#gapChars"><B>gapChars</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils"><B>GapFrequency</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/GapFrequency.html#GapFrequency()"><B>GapFrequency()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils">GapFrequency</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/IndelFilter.html#generate()"><B>generate()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils">IndelFilter</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#get(byte[])"><B>get(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Gets the child node associated with the given sequence, if it exists
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#get(byte[])"><B>get(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>Gets the child node associated with the given sequence, if it exists
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAllDownstreamNodes()"><B>getAllDownstreamNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Gets a List<RonPSTNode> of all nodes in the subtree starting from this node, including this node.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getAllUpstreamNodes()"><B>getAllUpstreamNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets a List<RonPSTNode> of all nodes in the subtree starting from this node, including this node.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAlphabet()"><B>getAlphabet()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Returns the alphabet of this SequenceSpectrum object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getAlphabet()"><B>getAlphabet()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Returns the alphabet of this SequenceSpectrum object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getAlphabet()"><B>getAlphabet()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Returns the alphabet of this SequenceSpectrum object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getAncestryList()"><B>getAncestryList()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getArray()"><B>getArray()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>Returns an array of the counts.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#getArray()"><B>getArray()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>Returns an array of the counts.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getArraySum()"><B>getArraySum()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getAverageGenomeSize()"><B>getAverageGenomeSize()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAvgDepth()"><B>getAvgDepth()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getAvgDepth()"><B>getAvgDepth()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getBackoffPrior()"><B>getBackoffPrior()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getBaseSpectrum()"><B>getBaseSpectrum()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Gets the base spectrum from which all other spectra are derived
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getBaseSpectrumRaw()"><B>getBaseSpectrumRaw()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Just get the baseSpectrum as is, with no synchronization or lazy initialization
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#getCharactersRead()"><B>getCharactersRead()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getCharactersRead()"><B>getCharactersRead()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#getCharactersRead()"><B>getCharactersRead()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>Returns the total number of nucleotides read
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#getCharactersRead()"><B>getCharactersRead()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#getCharactersRead()"><B>getCharactersRead()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getChild(byte)"><B>getChild(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Gets the child node associated with the given sequence, if it exists
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getChild(byte)"><B>getChild(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getChildren()"><B>getChildren()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getDescendant(byte[])"><B>getDescendant(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>gets the node associated with the the given sequence, or null if that node does not exist
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getDescendant(byte[])"><B>getDescendant(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getDesiredLength()"><B>getDesiredLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#getEmpty()"><B>getEmpty()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#getEmpty()"><B>getEmpty()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</A>
<DD>Returns an empty SequenceSpectrum of the appropriate type.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getExclusiveLabel()"><B>getExclusiveLabel()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getExclusiveLabel()"><B>getExclusiveLabel()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getExclusiveLabel()"><B>getExclusiveLabel()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getFactory()"><B>getFactory()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getFirstWordProvider()"><B>getFirstWordProvider()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html#getFirstWords(int)"><B>getFirstWords(int)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html" title="interface in edu.berkeley.compbio.sequtils.strings">FirstWordProvider</A>
<DD>Returns a list of all the initial words of length k that are associated with this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getFirstWords(int)"><B>getFirstWords(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getFirstWords(int)"><B>getFirstWords(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getGenomeCount()"><B>getGenomeCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getId()"><B>getId()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Returns a String representing the sequence of symbols leading to this node in the tree.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getId()"><B>getId()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets the id of this node, which is just the suffix associated with this node represented as a byte[], here converted
 to String for the sake of the Clusterable interface.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getId()"><B>getId()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Returns a String identifying this object.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getIdBytes()"><B>getIdBytes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getIdBytes()"><B>getIdBytes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets the id of this node, which is just the suffix associated with this node represented as a byte[].
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getImmutableWeightedLabels()"><B>getImmutableWeightedLabels()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getImmutableWeightedLabels()"><B>getImmutableWeightedLabels()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getItemCount()"><B>getItemCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getItemCount()"><B>getItemCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getK()"><B>getK()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns the pattern length K that this Kcount handles (i.e.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getLength()"><B>getLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns the length of this sequence
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getLength()"><B>getLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Returns the length of this sequence
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getLevelOneArray()"><B>getLevelOneArray()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getLogProbs()"><B>getLogProbs()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getLongestPrefix(byte[])"><B>getLongestPrefix(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>gets the node associated with the longest available prefix of the given sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getLongestSuffix(byte[])"><B>getLongestSuffix(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>gets the node associated with the longest available suffix of the given sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getMaxDepth()"><B>getMaxDepth()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Careful-- this gets cached on the first call so will be invalid later if the tree is changed
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getMaxDepth()"><B>getMaxDepth()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getMaxDepth()"><B>getMaxDepth()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Returns the maximum length of substrings considered in computing this statistical model of the sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#getMixingProportion()"><B>getMixingProportion()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getMutableWeightedLabels()"><B>getMutableWeightedLabels()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getMutableWeightedLabels()"><B>getMutableWeightedLabels()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html#getMutuallyExclusiveLabels()"><B>getMutuallyExclusiveLabels()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">LabelProvidingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#getName()"><B>getName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Returns the species name of this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#getName()"><B>getName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Returns the species name of this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getName()"><B>getName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Returns the name of the FastaFileSet
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getName()"><B>getName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#getName()"><B>getName()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Returns the species name of this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getName()"><B>getName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#getNeedlemanWunsch()"><B>getNeedlemanWunsch()</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow()"><B>getNextWindow()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow(int)"><B>getNextWindow(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow()"><B>getNextWindow()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow(int)"><B>getNextWindow(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow()"><B>getNextWindow()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow(int)"><B>getNextWindow(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow()"><B>getNextWindow()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow(int)"><B>getNextWindow(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#getNucleotideFilter()"><B>getNucleotideFilter()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getNumberOfBins()"><B>getNumberOfBins()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns the number of bins, which is the number of possible patterns in this Kcount.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getNumLeaves()"><B>getNumLeaves()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getNumLeaves()"><B>getNumLeaves()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getNumNodes()"><B>getNumNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getNumNodes()"><B>getNumNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getOriginalSequenceLength()"><B>getOriginalSequenceLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns the length of the sequence that was scanned to produce this spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getOriginalSequenceLength()"><B>getOriginalSequenceLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Returns the number of samples on which this spectrum is based.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getOriginalSequenceLength()"><B>getOriginalSequenceLength()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getOriginalSequenceLength()"><B>getOriginalSequenceLength()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Returns the length of the sequence that was scanned to produce this spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getParent()"><B>getParent()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>Returns the parent of this Kcount.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getParentMetadata()"><B>getParentMetadata()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns the SequenceFragmentMetadata representing the parent sequence, a longer sequence that contains this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getProbs()"><B>getProbs()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Returns a mapping of symbols to child nodes, describing the possible transitions from this node to its children.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getProbs()"><B>getProbs()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Gets the probability distribution over symbols following the suffix specified by this node's id.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getRawSequence()"><B>getRawSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>getReader(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Returns a SequenceReader at the given section
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, int)"><B>getReader(SequenceFragmentMetadata, int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Returns a SequenceReader at the given section
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getReaderForSynchronizing()"><B>getReaderForSynchronizing()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getReaderForSynchronizing()"><B>getReaderForSynchronizing()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getResetReader()"><B>getResetReader()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getResetReader()"><B>getResetReader()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Returns the number of samples on which the base spectrum is based.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getRootMetadata()"><B>getRootMetadata()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns a String identifying the largest sequence to which this fragment belongs (i.e., the root af the parent
 hierarchy)
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getRootMetadata()"><B>getRootMetadata()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#getRootMetadata()"><B>getRootMetadata()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getRootMetadata()"><B>getRootMetadata()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#getRootMetadata()"><B>getRootMetadata()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getRootSequenceName()"><B>getRootSequenceName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns a String identifying the largest sequence to which this fragment belongs (i.e., the root af the parent
 hierarchy)
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getScanner()"><B>getScanner()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getSequenceFragmentMetadata(java.lang.String)"><B>getSequenceFragmentMetadata(String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getSequenceFragments()"><B>getSequenceFragments()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getSequenceName()"><B>getSequenceName()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns a String identifying this sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#getSmithWaterman()"><B>getSmithWaterman()</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getSourceId()"><B>getSourceId()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getSourceId()"><B>getSourceId()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getSpectrum(java.lang.Class, com.davidsoergel.dsutils.GenericFactory)"><B>getSpectrum(Class&lt;X&gt;, GenericFactory&lt;X&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Provides a SequenceSpectrum of the requested type.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getStartPosition()"><B>getStartPosition()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns the start position of this sequence, a 0-based index with respect to the parent sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getStartPositionFromRoot()"><B>getStartPositionFromRoot()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getSubtreeNodes()"><B>getSubtreeNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getSumOfCounts()"><B>getSumOfCounts()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns the the sum of the counts.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#getTarget()"><B>getTarget()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getTaxid()"><B>getTaxid()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Returns the NCBI taxid of the taxon to which this sequence belongs
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getTaxid()"><B>getTaxid()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTaxid()"><B>getTaxid()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Returns the total amount of sequence present in this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Returns the total amount of sequence present in this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>Returns the total amount of sequence available
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Returns the total amount of sequence present in all files
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Returns the total amount of sequence present in this reader
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getTotalSequence()"><B>getTotalSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getTranslatedSequence()"><B>getTranslatedSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getUnknownCount()"><B>getUnknownCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getUnknownCount()"><B>getUnknownCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getUnknownCount()"><B>getUnknownCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getUnknownCount()"><B>getUnknownCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getUnknownCount()"><B>getUnknownCount()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Returns the number of characters from the original sequence that were not counted because they were unknown
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getUpstreamNode(byte)"><B>getUpstreamNode(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Get the upstream node for the specified symbol.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getUpstreamNodes()"><B>getUpstreamNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Returns an array of upstream nodes for each symbol, indexed by the alphabet index.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getWeightedLabels()"><B>getWeightedLabels()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_H_"><!-- --></A><H2>
<B>H</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#hasNext()"><B>hasNext()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#hasParent()"><B>hasParent()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#hasParent()"><B>hasParent()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>HierarchicalSpectrum</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="type parameter in HierarchicalSpectrum">T</A> extends <A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#HierarchicalSpectrum()"><B>HierarchicalSpectrum()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_I_"><!-- --></A><H2>
<B>I</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#id"><B>id</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#idForSequence(byte[])"><B>idForSequence(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#incrementBy(T)"><B>incrementBy(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>updates this object by adding another one to it.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#incrementBy(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)"><B>incrementBy(MarkovTreeNode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#incrementBy(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>incrementBy(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#incrementBy(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>incrementBy(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode, double)"><B>incrementByWeighted(MarkovTreeNode, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.RonPST, double)"><B>incrementByWeighted(RonPST, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.SequenceFragment, double)"><B>incrementByWeighted(SequenceFragment, double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#incrementGenomeCount()"><B>incrementGenomeCount()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils"><B>IndelFilter</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>InferredMixturePST</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD><B>Deprecated.</B>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html#InferredMixturePST()"><B>InferredMixturePST()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html" title="class in edu.berkeley.compbio.sequtils.strings">InferredMixturePST</A>
<DD><B>Deprecated.</B>&nbsp;&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#init()"><B>init()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#init()"><B>init()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#init()"><B>init()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#init()"><B>init()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#init()"><B>init()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initMainBuffer(int)"><B>initMainBuffer(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initTranslationBuffer(int)"><B>initTranslationBuffer(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>IntKcount</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="type parameter in IntKcount">T</A> extends <A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Represents the counts of a set of given patterns appearing in a string.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#IntKcount()"><B>IntKcount()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>Creates a new Kcount instance.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>IntPseudocountAdder</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#IntPseudocountAdder()"><B>IntPseudocountAdder()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isAligned(byte[][], int, int, int, int)"><B>isAligned(byte[][], int, int, int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>Tell whether position pos1 in sequence seq2 is aligned to pos2 in seq2 in the given char array.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isGap(byte)"><B>isGap(byte)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#isIgnoreEdges()"><B>isIgnoreEdges()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#isLeaf()"><B>isLeaf()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#isLeaf()"><B>isLeaf()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Tells whether this node is a leaf or not.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isLeftJustified(byte[])"><B>isLeftJustified(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#isMatePair(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>isMatePair(SequenceFragmentMetadata, SequenceFragmentMetadata)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#iupacPattern(java.lang.String)"><B>iupacPattern(String)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_J_"><!-- --></A><H2>
<B>J</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#joinMatePair(edu.berkeley.compbio.sequtils.strings.SequenceFragment, edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>joinMatePair(SequenceFragment, SequenceFragment)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_K_"><!-- --></A><H2>
<B>K</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#k"><B>k</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>Kcount</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="type parameter in Kcount">T</A> extends <A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#Kcount()"><B>Kcount()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings"><B>KcountScanner</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Provides an interface for classes that can scan sequences to count the occurrences of substrings<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>KneserNeyPSASmoother</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#KneserNeyPSASmoother()"><B>KneserNeyPSASmoother()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_L_"><!-- --></A><H2>
<B>L</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#label"><B>label</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>LabelProvidingSFI</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>LabelSelectableSFI</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>LairdSaulPST</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html#LairdSaulPST(java.io.Reader)"><B>LairdSaulPST(Reader)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html" title="class in edu.berkeley.compbio.sequtils.strings">LairdSaulPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#lastSymbolForId(int)"><B>lastSymbolForId(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#leaf"><B>leaf</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#learn(double, double, double, int, edu.berkeley.compbio.sequtils.strings.SequenceSpectrum, com.davidsoergel.stats.DistributionProcessor)"><B>learn(double, double, double, int, SequenceSpectrum, DistributionProcessor&lt;RonPSA&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#length"><B>length</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>The total number of characters represented by this fragment, including "N" characters or whatever.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#logConditionalProbability(byte)"><B>logConditionalProbability(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Returns the conditional probability of the given symbol from this node.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#logConditionalProbabilityByAlphabetIndex(int)"><B>logConditionalProbabilityByAlphabetIndex(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils"><B>LogOddsSubstitutionMatrix</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>A substitution matrix with integer scores (in log probability units), such as a PAM or BLOSUM matrix.<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#LogOddsSubstitutionMatrix()"><B>LogOddsSubstitutionMatrix()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_M_"><!-- --></A><H2>
<B>M</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>MarkovTreeNode</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>A node in a tree where each transition is labeled by a byte and has a local probability (that is, conditional on the
 parent node).<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#MarkovTreeNode()"><B>MarkovTreeNode()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#MarkovTreeNode(byte[], byte[])"><B>MarkovTreeNode(byte[], byte[])</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Constructs a new MarkovTreeNode with the given identifier
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#minus(T)"><B>minus(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns a new object representing the difference between this one and the given argument.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#minus(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)"><B>minus(MarkovTreeNode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#minus(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>minus(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#minus(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>minus(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#multiplyBy(double)"><B>multiplyBy(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#multiplyBy(double)"><B>multiplyBy(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#multiplyBy(double)"><B>multiplyBy(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#multiplyBy(double)"><B>multiplyBy(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_N_"><!-- --></A><H2>
<B>N</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#newParent()"><B>newParent()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>Generates a new Kcount based on this one and stores it as the parent
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Returns the next section
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#next()"><B>next()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()"><B>next()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>Iterates through the list of sections in order.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#nextFullyLabelled()"><B>nextFullyLabelled()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNode(byte)"><B>nextNode(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>Gets the child node associated with the given sequence, if it exists
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNodeByAlphabetIndex(int)"><B>nextNodeByAlphabetIndex(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNodes()"><B>nextNodes()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html#nextRequiringLabel(java.lang.String)"><B>nextRequiringLabel(String)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">LabelSelectableSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()"><B>nextShuffled()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#nextShuffled()"><B>nextShuffled()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#nextShuffled()"><B>nextShuffled()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()"><B>nextShuffled()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>Iterates through the list of sections in random order, providing each section exactly once
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#normalize()"><B>normalize()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Normalize the underlying probability distribution
<DT><A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils"><B>NotEnoughSequenceException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>Used when there is insufficient sequence in a reader for the next operation, such as the end of section encountered,
 or being inside a header.<DT><A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.String)"><B>NotEnoughSequenceException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.Exception)"><B>NotEnoughSequenceException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.Exception, java.lang.String)"><B>NotEnoughSequenceException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#numberOfBins"><B>numberOfBins</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/IndelFilter.html#numChars()"><B>numChars()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils">IndelFilter</A>
<DD>return the number of nucleotides to add or remove at a given nucletodie position.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numChars(byte[])"><B>numChars(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGapClusters(byte[])"><B>numGapClusters(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGaps(byte[])"><B>numGaps(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGaps(char[])"><B>numGaps(char[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numNonGaps(byte[])"><B>numNonGaps(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_O_"><!-- --></A><H2>
<B>O</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#originalSequenceLength"><B>originalSequenceLength</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#overlaps(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>overlaps(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Tells whether this fragment overlaps with another.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#overlaps(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>overlaps(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_P_"><!-- --></A><H2>
<B>P</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#padEndWithGaps(byte[], int)"><B>padEndWithGaps(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#padStartWithGaps(byte[], int)"><B>padStartWithGaps(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils"><B>PairAlignedDnaDist</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#PairAlignedDnaDist(edu.berkeley.compbio.sequtils.DnaDist, edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner)"><B>PairAlignedDnaDist(DnaDist, DynamicProgrammingPairwiseAligner)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#parent"><B>parent</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#parentMetadata"><B>parentMetadata</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#peek()"><B>peek()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#plus(T)"><B>plus(T)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>Returns a new object representing the sum of this one and the given argument.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#plus(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)"><B>plus(MarkovTreeNode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#plus(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>plus(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#plus(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>plus(SequenceFragment)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#prefixForId(int)"><B>prefixForId(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#prefixId(int)"><B>prefixId(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#probs"><B>probs</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#process(com.davidsoergel.stats.DoubleArrayContainer)"><B>process(DoubleArrayContainer)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#process(com.davidsoergel.stats.IntArrayContainer)"><B>process(IntArrayContainer)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPSA)"><B>process(RonPSA)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPSA)"><B>process(RonPSA)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPST)"><B>process(RonPST)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_R_"><!-- --></A><H2>
<B>R</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()"><B>randomSectionFragment()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionFragment()"><B>randomSectionFragment()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionFragment()"><B>randomSectionFragment()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()"><B>randomSectionFragment()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionLengthWeighted()"><B>randomSectionLengthWeighted()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionLengthWeighted()"><B>randomSectionLengthWeighted()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionLengthWeighted()"><B>randomSectionLengthWeighted()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()"><B>randomSectionLengthWeighted()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>Get a random section from the list, selected uniformly from the complete sections in the list, weighted by their
 lengths.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader"><B>RandomSectionList</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</A><DD>Provides randomly generated sequence implementing SequenceReader<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#RandomSectionList()"><B>RandomSectionList()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>RandomSectionSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads sections.<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#RandomSectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>RandomSectionSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionUniform()"><B>randomSectionUniform()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionUniform()"><B>randomSectionUniform()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionUniform()"><B>randomSectionUniform()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()"><B>randomSectionUniform()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>RandomSequentialMatePairSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads random mate-pairs<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#RandomSequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>RandomSequentialMatePairSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#RandomSequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList, int)"><B>RandomSequentialMatePairSFI(SectionList, int)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read()"><B>read()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Returns the next nucleotide
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read(byte[], int)"><B>read(byte[], int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Reads nucleotides into a byte array
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read()"><B>read()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Returns the next nucleotide
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read(byte[], int)"><B>read(byte[], int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Reads nucleotides into a byte array
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()"><B>read()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read(byte[], int)"><B>read(byte[], int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read()"><B>read()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read(byte[], int)"><B>read(byte[], int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read(int)"><B>read(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#read()"><B>read()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Returns the next nucleotide
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[], int)"><B>read(byte[], int)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Reads nucleotides into a byte array
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#readAt(byte[], long, int)"><B>readAt(byte[], long, int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Reads a section from the files
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#readTranslated()"><B>readTranslated()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#readTranslated()"><B>readTranslated()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()"><B>readTranslated()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()"><B>readTranslated()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#realign(com.davidsoergel.dsutils.collections.OrderedPair)"><B>realign(OrderedPair&lt;byte[], byte[]&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#releaseCachedResources()"><B>releaseCachedResources()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Closes all the open filehandles
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#remove()"><B>remove()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</A>
<DD>Unsupported, as all iteration is online
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#remove()"><B>remove()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#removeOverlaps(java.util.Collection)"><B>removeOverlaps(Collection&lt;SequenceFragment&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Careful: this does not propagate labels to the subsidiary fragments, because they may or may not apply.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#replaceGaps(byte[])"><B>replaceGaps(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanRawSequence()"><B>rescanRawSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#rescanSpectrum()"><B>rescanSpectrum()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanSpectrum()"><B>rescanSpectrum()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanTranslatedSequence()"><B>rescanTranslatedSequence()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>reset the stream to the beginning.
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>resets the base reader.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#reset()"><B>reset()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#reset()"><B>reset()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>reset the stream to the beginning.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()"><B>reset()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A>
<DD>reset the stream to the beginning.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#reverseComplement(byte[])"><B>reverseComplement(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPSA</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#RonPSA()"><B>RonPSA()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPSANode</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Extends a MarkovTreeNode with backlinks, for making make a Probabilistic Suffix Automaton (PSA).<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#RonPSANode()"><B>RonPSANode()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#RonPSANode(byte[], byte[])"><B>RonPSANode(byte[], byte[])</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>Constructs a new MarkovTreeNode with the given identifier
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPSASmoother</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#RonPSASmoother()"><B>RonPSASmoother()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPST</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>A probabilistic suffix tree, as described in
 <p/>
 D.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#RonPST()"><B>RonPST()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#RonPST(double, double, double, int, edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>RonPST(double, double, double, int, SequenceSpectrum)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Constructs a new Probabilistic Suffix Tree according to the Learn-PST algorithm (Ron et al 1996 p.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPSTNode</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>A node of a Probabilistic Suffix Tree, where each symbol is a byte.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#RonPSTNode()"><B>RonPSTNode()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#RonPSTNode(byte[], byte[])"><B>RonPSTNode(byte[], byte[])</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Constructs a new RonPSTNode with the given identifier
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>RonPSTSmoother</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#RonPSTSmoother()"><B>RonPSTSmoother()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotate(byte[], int, int)"><B>rotate(byte[], int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotate(byte[], int, int, int)"><B>rotate(byte[], int, int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>Rotates an array slice in place within the larger array.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotateUpdatingGaps(int[], int, int)"><B>rotateUpdatingGaps(int[], int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotateUpdatingGaps(int[], int, int, int)"><B>rotateUpdatingGaps(int[], int, int, int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#runPhaseInit()"><B>runPhaseInit()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#runPhaseInit()"><B>runPhaseInit()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_S_"><!-- --></A><H2>
<B>S</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#sample(byte[])"><B>sample(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#sample(int)"><B>sample(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Chooses a random string according to the conditional probabilities of symbols.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#sample(byte[])"><B>sample(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#sample(int)"><B>sample(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Chooses a random string according to the conditional probabilities of symbols.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#sample(byte[])"><B>sample(byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#sample(int)"><B>sample(int)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Chooses a random string according to the conditional probabilities of symbols.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>ScanningSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#ScanningSFI()"><B>ScanningSFI()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#ScanningSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>ScanningSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>scanSequence(SequenceFragment)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">KcountScanner</A>
<DD>Scans a sequence to count pattern frequencies.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment, byte[])"><B>scanSequence(SequenceFragment, byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">KcountScanner</A>
<DD>Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)"><B>scanSequence(SequenceFragment)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</A>
<DD>Scans a sequence to count pattern frequencies.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment, byte[])"><B>scanSequence(SequenceFragment, byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</A>
<DD>Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.
<DT><A HREF="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html#score(byte, byte, boolean)"><B>score(byte, byte, boolean)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html" title="interface in edu.berkeley.compbio.sequtils">AffineSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#score(byte, byte)"><B>score(byte, byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html#score(byte, byte, boolean)"><B>score(byte, byte, boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">SimpleSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#scoreAgainstGap(byte)"><B>scoreAgainstGap(byte)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>SectionDecomposingSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads section lists<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#SectionDecomposingSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>SectionDecomposingSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#sectionList"><B>sectionList</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader"><B>SectionList</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</A><DD>A list of sequence fragments, represented only by their metadata.<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>SectionListBasedSequenceFragmentIterator</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#SectionListBasedSequenceFragmentIterator(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>SectionListBasedSequenceFragmentIterator(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#SectionListBasedSequenceFragmentIterator()"><B>SectionListBasedSequenceFragmentIterator()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>SectionSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads sections.<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#SectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>SectionSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</A>
<DD>Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(long)"><B>seek(long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>seek(SequenceFragmentMetadata)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><B>seek(SequenceFragmentMetadata, long)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#seqIndexToColumn(byte[], int)"><B>seqIndexToColumn(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>Return the column (0-based) in which the n'th (0-based) non-gap character appears.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceArrayException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.String)"><B>SequenceArrayException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.Exception)"><B>SequenceArrayException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.Exception, java.lang.String)"><B>SequenceArrayException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceArrayUtils</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#SequenceArrayUtils()"><B>SequenceArrayUtils()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceError</B></A> - Error in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.String)"><B>SequenceError(String)</B></A> - 
Constructor for error edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.Exception)"><B>SequenceError(Exception)</B></A> - 
Constructor for error edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.Exception, java.lang.String)"><B>SequenceError(Exception, String)</B></A> - 
Constructor for error edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.String)"><B>SequenceException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.Exception)"><B>SequenceException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.Exception, java.lang.String)"><B>SequenceException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#sequenceForId(int)"><B>sequenceForId(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>SequenceFragment</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Manages information about a sequence fragment, including its metadata and any statistics that have been calculated
 from the sequence, in the form of SequenceSpectra.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, long, long, edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)"><B>SequenceFragment(SequenceFragmentMetadata, String, long, long, SequenceSpectrumScanner)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Constructs a new SequenceFragment by specifying its coordinates with respect to a containing parent sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, long, edu.berkeley.compbio.sequtils.SequenceReader, long, edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)"><B>SequenceFragment(SequenceFragmentMetadata, String, long, SequenceReader, long, SequenceSpectrumScanner)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, long, edu.berkeley.compbio.sequtils.SequenceReader, long, edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner, long)"><B>SequenceFragment(SequenceFragmentMetadata, String, long, SequenceReader, long, SequenceSpectrumScanner, long)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Convenience constructor, creates a new SequenceFragment by applying a SequenceSpectrumScanner to an input stream,
 and using the resulting SequenceSpectrum as the base spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(byte[], edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)"><B>SequenceFragment(byte[], SequenceSpectrumScanner)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>SequenceFragmentIterator</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceFragmentMetadata</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="type parameter in SequenceFragmentMetadata">C</A> extends <A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>Provides a container for information about a particular sequence.<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#SequenceFragmentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, java.lang.Integer, int)"><B>SequenceFragmentMetadata(SequenceFragmentMetadata, String, Integer, int)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Constructs a new SequenceFragmentMetadata by specifying its start position with respect to a containing parent
 sequence, but with an unknown length.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#SequenceFragmentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, java.lang.String, java.lang.Integer, long, long)"><B>SequenceFragmentMetadata(SequenceFragmentMetadata, String, Integer, long, long)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Constructs a new SequenceFragmentMetadata by specifying its coordinates with respect to a containing parent
 sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#sequenceName"><B>sequenceName</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils"><B>SequenceReader</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>Provides an interface for a buffered sequence reader, to return nucleotides and process parsers or mutation
 transparently.<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils"><B>SequenceRuntimeException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.String)"><B>SequenceRuntimeException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.Exception)"><B>SequenceRuntimeException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.Exception, java.lang.String)"><B>SequenceRuntimeException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings"><B>SequenceSpectrum</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="type parameter in SequenceSpectrum">T</A> extends <A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>&gt; - Interface in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Represents a statistical model of a sequence of bytes.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>SequenceSpectrumException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Thrown when something involving a SequenceSpectrum goes wrong, such as when a requested spectrum cannot be computed.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.String)"><B>SequenceSpectrumException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.Exception)"><B>SequenceSpectrumException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.Exception, java.lang.String)"><B>SequenceSpectrumException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>SequenceSpectrumRuntimeException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Thrown when something involving a SequenceSpectrum goes wrong, such as when a requested spectrum cannot be computed.<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.String)"><B>SequenceSpectrumRuntimeException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.Exception)"><B>SequenceSpectrumRuntimeException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.Exception, java.lang.String)"><B>SequenceSpectrumRuntimeException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings"><B>SequenceSpectrumScanner</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>Provides an interface for classes that can scan sequences to count the occurrences of substrings<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>SequentialMatePairSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads sequential mate-pairs<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#SequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>SequentialMatePairSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setAlphabet(byte[])"><B>setAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setAlphabet(byte[])"><B>setAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#setAlphabet(byte[])"><B>setAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Sets the array of valid symbols.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setBacklinksUsingRoot(edu.berkeley.compbio.sequtils.strings.RonPSANode, java.util.Queue)"><B>setBacklinksUsingRoot(RonPSANode, Queue&lt;RonPSANode&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setBackoffPrior(edu.berkeley.compbio.sequtils.strings.RonPSANode)"><B>setBackoffPrior(RonPSANode)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</A>
<DD>This method assumes that it has not been run before, and that all nodes at higher levels in the tree have already
 been proceesed.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setBaseSpectrum(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>setBaseSpectrum(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Sets the base spectrum from which all other spectra are derived
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>setFragmentLength(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setId(byte[])"><B>setId(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#setId(byte[])"><B>setId(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Sets the id of this node, which is just the suffix associated with this node represented as a byte[].
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setIgnoreEdges(boolean)"><B>setIgnoreEdges(boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setIgnoreEdges(boolean)"><B>setIgnoreEdges(boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Specify whether or not this sequence spectrum attempts to account for edge effects, i.e.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setIgnoreEdges(boolean)"><B>setIgnoreEdges(boolean)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#setIgnoreEdges(boolean)"><B>setIgnoreEdges(boolean)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Specify whether or not this sequence spectrum attempts to account for edge effects, i.e.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setImmutable()"><B>setImmutable()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setImmutable()"><B>setImmutable()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Declare that this spectrum should not change anymore.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#setImmutable()"><B>setImmutable()</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Declare that this spectrum should not change anymore.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#setLabel(java.lang.String)"><B>setLabel(String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setLabel(java.lang.String)"><B>setLabel(String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setLabel(java.lang.String)"><B>setLabel(String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setLength(int)"><B>setLength(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Sets the length of this sequence
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setOriginalSequenceLength(long)"><B>setOriginalSequenceLength(long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setParentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><B>setParentMetadata(SequenceFragmentMetadata)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Sets the SequenceFragmentMetadata representing the parent sequence, a longer sequence that contains this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#setResults(com.davidsoergel.trees.dhtpn.SerializableDoubleHierarchicalTypedProperties)"><B>setResults(SerializableDoubleHierarchicalTypedProperties&lt;?&gt;)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</A>
<DD>Frees up memory, filehandles, etc.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#setResults(com.davidsoergel.trees.dhtpn.SerializableDoubleHierarchicalTypedProperties)"><B>setResults(SerializableDoubleHierarchicalTypedProperties&lt;?&gt;)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setSequenceName(java.lang.String)"><B>setSequenceName(String)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Sets a String identifying this sequence
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setStartPosition(long)"><B>setStartPosition(long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>Sets the start position of this sequence, a 0-based index with respect to the parent sequence.
<DT><A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#setTranslationAlphabet(byte[])"><B>setTranslationAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</A>
<DD>Set the character mapping to be used when reading nucleotides.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#setTranslationAlphabet(byte[])"><B>setTranslationAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])"><B>setTranslationAlphabet(byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>
<DD>Set the character mapping to be used when reading nucleotides.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setTranslationAlphabet(byte[])"><B>setTranslationAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationAlphabet(byte[])"><B>setTranslationAlphabet(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>Set the character mapping to be used when reading nucleotides.
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationInvalid()"><B>setTranslationInvalid()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setUnknownCount(long)"><B>setUnknownCount(long)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>ShuffledSectionSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>Provides a SequenceFragmentIterator that reads sections.<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#ShuffledSectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)"><B>ShuffledSectionSFI(SectionList)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils"><B>SimpleSubstitutionMatrix</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html#SimpleSubstitutionMatrix(float, float, float, float)"><B>SimpleSubstitutionMatrix(float, float, float, float)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">SimpleSubstitutionMatrix</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings"><B>SimplexVectorTargetAndProportion</B></A>&lt;<A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="type parameter in SimplexVectorTargetAndProportion">T</A>&gt; - Class in <A HREF="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#SimplexVectorTargetAndProportion(com.davidsoergel.stats.Multinomial, com.davidsoergel.stats.Multinomial)"><B>SimplexVectorTargetAndProportion(Multinomial&lt;T&gt;, Multinomial&lt;T&gt;)</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</A>
<DD>Constructs a new RelativeMarkovTreeNode by comparing one MarkovTreeNode with another and representing the difference
 as a vector on the simplex.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#smoothFactor"><B>smoothFactor</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#smoothFactor"><B>smoothFactor</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#smoothFactor"><B>smoothFactor</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</A>
<DD>Tests whether the given sequence statistics are equivalent to this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</A>
<DD>Test whether the given sequence statistics are equivalent to this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Test whether the given sequence statistics are equivalent to this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Test whether the given sequence statistics are equivalent to this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>Test whether this sequence has statistics matching the given spectrum.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)"><B>spectrumEquals(SequenceSpectrum)</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Test whether the given sequence statistics are equivalent to this one.
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#spectrumScanner"><B>spectrumScanner</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#startPosition"><B>startPosition</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startPositionOfGapCluster(byte[], int)"><B>startPositionOfGapCluster(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithGaps(byte[], int)"><B>startsWithGaps(byte[], int)</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithIUPAC(byte[], byte[])"><B>startsWithIUPAC(byte[], byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithIUPAC(byte[], int, byte[])"><B>startsWithIUPAC(byte[], int, byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#stripGaps(byte[][])"><B>stripGaps(byte[][])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>Strip all gaps, left-justifying the sequences.
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#stripGaps(byte[])"><B>stripGaps(byte[])</B></A> - 
Static method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>Push all the gaps to the end of the array, in place
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>StubSFI</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#StubSFI()"><B>StubSFI()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils"><B>SubstitutionFilter</B></A> - Interface in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>A filter that mutates nucleotides one at a time (i.e., without taking context into account).<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#suffixId(int)"><B>suffixId(int)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_T_"><!-- --></A><H2>
<B>T</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#theDerivedSpectra"><B>theDerivedSpectra</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#theScanner"><B>theScanner</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#times(double)"><B>times(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#times(double)"><B>times(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#times(double)"><B>times(double)</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#toLongString()"><B>toLongString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#toLongString()"><B>toLongString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html#toString()"><B>toString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#toString()"><B>toString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#toString()"><B>toString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#toString()"><B>toString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>
<DD>Returns the name of the FastaFileSet
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#toString()"><B>toString()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<A HREF="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#totalFragment()"><B>totalFragment()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<A HREF="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</A>
<DD>Note this is partly redundant with RepeatingAggregatingSFI
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#totalProbability(byte[])"><B>totalProbability(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>Computes the probability of generating the given sequence under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html#totalProbability(byte[])"><B>totalProbability(byte[])</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</A>
<DD>Computes the probability of generating the given sequence under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#totalProbability(byte[])"><B>totalProbability(byte[])</B></A> - 
Method in interface edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A>
<DD>Computes the probability of generating the given sequence under the model.
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><B>TranslatingSequenceReader</B></A> - Class in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#TranslatingSequenceReader()"><B>TranslatingSequenceReader()</B></A> - 
Constructor for class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils"><B>TranslationException</B></A> - Exception in <A HREF="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A><DD>&nbsp;<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.String)"><B>TranslationException(String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.Exception)"><B>TranslationException(Exception)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.Exception, java.lang.String)"><B>TranslationException(Exception, String)</B></A> - 
Constructor for exception edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</A>
<DD>&nbsp;
</DL>
<HR>
<A NAME="_U_"><!-- --></A><H2>
<B>U</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#uniformPseudoCount"><B>uniformPseudoCount</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#uniformPseudoCount"><B>uniformPseudoCount</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#UNKNOWN_BYTE"><B>UNKNOWN_BYTE</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#UNKNOWN_LENGTH"><B>UNKNOWN_LENGTH</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#UNKNOWN_NUMBER_OF_SAMPLES"><B>UNKNOWN_NUMBER_OF_SAMPLES</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#UNKNOWN_SUM_OF_COUNTS"><B>UNKNOWN_SUM_OF_COUNTS</B></A> - 
Static variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#unknownCount"><B>unknownCount</B></A> - 
Variable in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html#unknownProportion()"><B>unknownProportion()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#updateLogProbs()"><B>updateLogProbs()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#updateLogProbs()"><B>updateLogProbs()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Takes the logs of all the probabilities at this node and caches them.
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#updateLogProbsRecursive()"><B>updateLogProbsRecursive()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</A>
<DD>&nbsp;
<DT><A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#updateLogProbsRecursive()"><B>updateLogProbsRecursive()</B></A> - 
Method in class edu.berkeley.compbio.sequtils.strings.<A HREF="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</A>
<DD>Takes the logs of all the probabilities at this node and caches them; then follows the upstream transitions and
 repeats recursively.
</DL>
<HR>
<A NAME="_V_"><!-- --></A><H2>
<B>V</B></H2>
<DL>
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html#valueOf(java.lang.String)"><B>valueOf(String)</B></A> - 
Static method in enum edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackBegin</A>
<DD>Returns the enum constant of this type with the specified name.
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html#valueOf(java.lang.String)"><B>valueOf(String)</B></A> - 
Static method in enum edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackEnd</A>
<DD>Returns the enum constant of this type with the specified name.
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html#values()"><B>values()</B></A> - 
Static method in enum edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackBegin</A>
<DD>Returns an array containing the constants of this enum type, in
the order they are declared.
<DT><A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html#values()"><B>values()</B></A> - 
Static method in enum edu.berkeley.compbio.sequtils.<A HREF="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackEnd</A>
<DD>Returns an array containing the constants of this enum type, in
the order they are declared.
</DL>
<HR>
<A HREF="#_A_">A</A> <A HREF="#_B_">B</A> <A HREF="#_C_">C</A> <A HREF="#_D_">D</A> <A HREF="#_E_">E</A> <A HREF="#_F_">F</A> <A HREF="#_G_">G</A> <A HREF="#_H_">H</A> <A HREF="#_I_">I</A> <A HREF="#_J_">J</A> <A HREF="#_K_">K</A> <A HREF="#_L_">L</A> <A HREF="#_M_">M</A> <A HREF="#_N_">N</A> <A HREF="#_O_">O</A> <A HREF="#_P_">P</A> <A HREF="#_R_">R</A> <A HREF="#_S_">S</A> <A HREF="#_T_">T</A> <A HREF="#_U_">U</A> <A HREF="#_V_">V</A> 

<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Package</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Class</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <FONT CLASS="NavBarFont1">Use</FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./overview-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Index</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="./help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="./index.html?index-all.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="index-all.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="./allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="./allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.
</BODY>
</HTML>
