<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_60) on Tue Jul 08 13:00:18 EDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>Index (sequtils 0.92-SNAPSHOT API)</title>
<meta name="date" content="2014-07-08">
<link rel="stylesheet" type="text/css" href="./stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Index (sequtils 0.92-SNAPSHOT API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="./overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="./overview-tree.html">Tree</a></li>
<li><a href="./deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="./help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="./index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="./allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="contentContainer"><a href="#_A_">A</a>&nbsp;<a href="#_B_">B</a>&nbsp;<a href="#_C_">C</a>&nbsp;<a href="#_D_">D</a>&nbsp;<a href="#_E_">E</a>&nbsp;<a href="#_F_">F</a>&nbsp;<a href="#_G_">G</a>&nbsp;<a href="#_H_">H</a>&nbsp;<a href="#_I_">I</a>&nbsp;<a href="#_J_">J</a>&nbsp;<a href="#_K_">K</a>&nbsp;<a href="#_L_">L</a>&nbsp;<a href="#_M_">M</a>&nbsp;<a href="#_N_">N</a>&nbsp;<a href="#_O_">O</a>&nbsp;<a href="#_P_">P</a>&nbsp;<a href="#_R_">R</a>&nbsp;<a href="#_S_">S</a>&nbsp;<a href="#_T_">T</a>&nbsp;<a href="#_U_">U</a>&nbsp;<a href="#_V_">V</a>&nbsp;<a name="_A_">
<!--   -->
</a>
<h2 class="title">A</h2>
<dl>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">AbstractSequenceFragmentIterator</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides an interface for an Iterator of Kcounts</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#AbstractSequenceFragmentIterator()">AbstractSequenceFragmentIterator()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#add(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">add(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#add(byte[])">add(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given sequence, creating it (and nodes along the way) as needed</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addAll(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet)">addAll(FastaFileSet)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#addChild(byte)">addChild(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given symbol, creating it first if needed.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#addChild(byte)">addChild(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given symbol, creating it first if needed.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addDirectory(java.io.File)">addDirectory(File)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#addFile(java.io.File)">addFile(File)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#addPseudocounts()">addPseudocounts()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#addUnknown()">addUnknown()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Adds an "unknown" sample to this kcount, indicating that a character was consumed without incrementing any counter.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#addUpstreamNode(byte[])">addUpstreamNode(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets the upstream node associated with the given suffix starting from this node, creating it (and nodes along the
 way) as needed</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#addUpstreamNode(byte)">addUpstreamNode(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets the upstream node associated with the given symbol starting from this node, creating it first if needed.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html" title="interface in edu.berkeley.compbio.sequtils"><span class="strong">AffineSubstitutionMatrix</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">AggregateSequenceFragment</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#AggregateSequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20java.util.Collection)">AggregateSequenceFragment(SequenceFragmentMetadata, String, Collection&lt;SequenceFragment&gt;)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>
<div class="block">Constructs a new SequenceFragment by specifying its coordinates with respect to a containing parent sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#AggregateSequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20edu.berkeley.compbio.sequtils.strings.SequenceFragment)">AggregateSequenceFragment(SequenceFragmentMetadata, String, SequenceFragment)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#align(byte[],%20byte[])">align(byte[], byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#alphabet">alphabet</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#alphabet">alphabet</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#appendString(java.util.Formatter,%20java.lang.String)">appendString(Formatter, String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#appendString(java.util.Formatter,%20java.lang.String)">appendString(Formatter, String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#appendString(java.util.Formatter,%20java.lang.String)">appendString(Formatter, String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Appends a string representation of this node to the given formatter, indenting as needed based on the hierarchy
 level.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#applyMask(boolean[],%20byte[],%20byte)">applyMask(boolean[], byte[], byte)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">copy a slice ending at the given position (exclusive!) of whatever width is needed to obtain the requested number of
 non-gap characters.</div>
</dd>
</dl>
<a name="_B_">
<!--   -->
</a>
<h2 class="title">B</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#buf">buf</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#bufPosition">bufPosition</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">ByteArraySequenceReader</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#ByteArraySequenceReader(byte[])">ByteArraySequenceReader(byte[])</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#ByteArraySequenceReader(java.lang.String)">ByteArraySequenceReader(String)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_C_">
<!--   -->
</a>
<h2 class="title">C</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#checkAvailable()">checkAvailable()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#children">children</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>
<div class="block">Clones this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>
<div class="block">Clone this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>
<div class="block">Clone this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Performs a deep copy of this node.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Clone this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#clone()">clone()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Clone this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#clone()">clone()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Closes the reader, and all sub-readers</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Closes the reader, and all sub-readers</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#close()">close()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>
<div class="block">Closes the SequenceFragmentIterator and all sub-readers</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#close()">close()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Closes the reader, and all sub-readers</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#close()">close()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#columnToSeqIndex(byte[],%20int)">columnToSeqIndex(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#compareTo(C)">compareTo(C)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Compares this object with the specified object for order.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#completionProcessor">completionProcessor</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte)">conditionalProbability(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte[],%20byte[])">conditionalProbability(byte[], byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbability(byte,%20byte[])">conditionalProbability(byte, byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#conditionalProbability(byte,%20byte[])">conditionalProbability(byte, byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Computes the conditional probability of generating a symbol given a prefix under the model, backing off to shorter
 prefixes as needed if the given prefix is not explicitly represented.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#conditionalProbability(byte,%20byte[])">conditionalProbability(byte, byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Computes the conditional probability of generating a symbol given a prefix under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalProbabilityByAlphabetIndex(int)">conditionalProbabilityByAlphabetIndex(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#conditionalsFrom(byte[])">conditionalsFrom(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Computes the conditional probability distribution of symbols given a prefix under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#conditionalsFrom(byte[])">conditionalsFrom(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Computes the conditional probability distribution of symbols given a prefix under the model, backing off to shorter
 prefixes as needed if the given prefix is not explicitly represented.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#conditionalsFrom(byte[])">conditionalsFrom(byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Computes the conditional probability distribution of symbols given a prefix under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copyNoGaps(byte[])">copyNoGaps(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#copyProbsFrom(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">copyProbsFrom(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Recursively fills out the tree so that each node has either a complete set of probabilities or none at all, and
 assigns conditional probabilities according to the given spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#copyProbsFromSpectrumRecursively(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">copyProbsFromSpectrumRecursively(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Recursively fills out the tree so that each node has either a complete set of children or none at all, and assigns
 conditional probabilities according to the given spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySlice(byte[],%20int,%20int)">copySlice(byte[], int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySliceUpToNNonGaps(byte[],%20int,%20int)">copySliceUpToNNonGaps(byte[], int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">copy a slice starting at the given position of whatever width is needed to obtain the requested number of non-gap
 characters</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copyWithout(byte[],%20byte)">copyWithout(byte[], byte)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#countChildren()">countChildren()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#counts">counts</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#counts">counts</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#countUpstreamNodes()">countUpstreamNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Counts the immediately upstream nodes that exist.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#currentSection">currentSection</a></span> - Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#currentSectionPosition">currentSectionPosition</a></span> - Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_D_">
<!--   -->
</a>
<h2 class="title">D</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#dataSum">dataSum</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#decrementBy(T)">decrementBy(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">updates this object by subtracting another one from it.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#decrementBy(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)">decrementBy(MarkovTreeNode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#decrementBy(edu.berkeley.compbio.sequtils.strings.RonPST)">decrementBy(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#decrementBy(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">decrementBy(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode,%20double)">decrementByWeighted(MarkovTreeNode, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.RonPST,%20double)">decrementByWeighted(RonPST, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#decrementByWeighted(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20double)">decrementByWeighted(SequenceFragment, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#desiredLengthUnknown()">desiredLengthUnknown()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#diagnostics()">diagnostics()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DnaDist.html#distanceFromTo(byte[],%20byte[])">distanceFromTo(byte[], byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/GapFrequency.html#distanceFromTo(byte[],%20byte[])">distanceFromTo(byte[], byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils">GapFrequency</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#distanceFromTo(byte[],%20byte[])">distanceFromTo(byte[], byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">DnaDist</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DnaDist.html#DnaDist(double,%20double,%20int)">DnaDist(double, double, int)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#doneLabelling()">doneLabelling()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#doneLabelling()">doneLabelling()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">DoubleKcount</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="type parameter in DoubleKcount">T</a> extends <a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Represents the counts of a set of given patterns appearing in a string.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#DoubleKcount(double[])">DoubleKcount(double[])</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>
<div class="block">Creates a new Kcount instance.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">DoublePseudocountAdder</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#DoublePseudocountAdder()">DoublePseudocountAdder()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">DynamicProgrammingPairwiseAligner</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#DynamicProgrammingPairwiseAligner(edu.berkeley.compbio.sequtils.AffineSubstitutionMatrix,%20edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner.TracebackBegin,%20edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner.TracebackEnd)">DynamicProgrammingPairwiseAligner(AffineSubstitutionMatrix, DynamicProgrammingPairwiseAligner.TracebackBegin, DynamicProgrammingPairwiseAligner.TracebackEnd)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils"><span class="strong">DynamicProgrammingPairwiseAligner.TracebackBegin</span></a> - Enum in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils"><span class="strong">DynamicProgrammingPairwiseAligner.TracebackEnd</span></a> - Enum in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_E_">
<!--   -->
</a>
<h2 class="title">E</h2>
<dl>
<dt><a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a> - package edu.berkeley.compbio.sequtils</dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a> - package edu.berkeley.compbio.sequtils.sequencefragmentiterator</dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a> - package edu.berkeley.compbio.sequtils.sequencereader</dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a> - package edu.berkeley.compbio.sequtils.sequencereader.fasta</dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a> - package edu.berkeley.compbio.sequtils.strings</dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#endsWithGaps(byte[],%20int)">endsWithGaps(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#ensureAllParentsExist()">ensureAllParentsExist()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>
<div class="block">Recursively generalize thisKcount, creating a chain of "parents" until no further generalization is possible</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#equalValue(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">equalValue(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#equalValue(T)">equalValue(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>
<div class="block">Tests whether the given pattern counts are equal to those stored here.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#equalValue(T)">equalValue(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>
<div class="block">Test whether the given pattern counts are equal to those stored here.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#equalValue(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)">equalValue(MarkovTreeNode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Recursively tests the deep equality of two PSTNodes.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#equalValue(edu.berkeley.compbio.sequtils.strings.RonPST)">equalValue(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#equalValue(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">equalValue(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Test whether the given object is the same as this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</a></dt>
<dd>
<div class="block">Note this is partly redundant with RepeatingAggregatingSFI</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>
<div class="block">provides the next item from the iterator without actually consuming it</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</a></dt>
<dd>
<div class="block">Note this is partly redundant with RepeatingAggregatingSFI</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#estimatedTotalSamples()">estimatedTotalSamples()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_F_">
<!--   -->
</a>
<h2 class="title">F</h2>
<dl>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">FastaFileSet</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a></dt>
<dd>
<div class="block">Fasta File Set, useful in treating multiple fasta files as one</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.lang.Integer)">FastaFileSet(Integer)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.lang.String)">FastaFileSet(String)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.io.File)">FastaFileSet(File)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Initializes a file set from a file or directory</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.util.Collection)">FastaFileSet(Collection&lt;File&gt;)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Initializes a file set from a file or directory</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.io.File,%20java.lang.Integer)">FastaFileSet(File, Integer)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#FastaFileSet(java.util.Collection,%20java.lang.Integer)">FastaFileSet(Collection&lt;FastaFileSet&gt;, Integer)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span></div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">FastaParser</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a></dt>
<dd>
<div class="block">Fasta File Parser</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet,%20boolean)">FastaParser(FastaFileSet, boolean)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Initializes a parser from a fileset</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet,%20int,%20boolean)">FastaParser(FastaFileSet, int, boolean)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Initializes a parser from a fileset and with a specific buffer size</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">FastaParserSet</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a></dt>
<dd>
<div class="block">Creates a HashSet of FastaParsers</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.util.Collection)">FastaParserSet(Collection&lt;File&gt;)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.io.File)">FastaParserSet(File)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html#FastaParserSet(java.io.File,%20int)">FastaParserSet(File, int)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParserSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#filename">filename</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte)">filter(byte)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</a></dt>
<dd>
<div class="block">Mutate the provided nucleotide and return it.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte[])">filter(byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</a></dt>
<dd>
<div class="block">Mutate each nucleotide in the provided byte buffer, in place.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html#filter(byte[],%20int)">filter(byte[], int)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils">SubstitutionFilter</a></dt>
<dd>
<div class="block">Mutate the first n nucleotides in the provided byte buffer, in place.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">FilterException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.String)">FilterException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.Exception)">FilterException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilterException.html#FilterException(java.lang.Exception,%20java.lang.String)">FilterException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader"><span class="strong">FilteringSectionList</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#FilteringSectionList(edu.berkeley.compbio.sequtils.sequencereader.SectionList,%20edu.berkeley.compbio.sequtils.SubstitutionFilter)">FilteringSectionList(SectionList, SubstitutionFilter)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">FilteringSequenceReader</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#FilteringSequenceReader(edu.berkeley.compbio.sequtils.SequenceReader,%20edu.berkeley.compbio.sequtils.SubstitutionFilter)">FilteringSequenceReader(SequenceReader, SubstitutionFilter)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#fireUpdated(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">fireUpdated(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Indicates that the sequence statistics have changed in some way, such that any derived statistics must be
 recomputed.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html" title="interface in edu.berkeley.compbio.sequtils.strings"><span class="strong">FirstWordProvider</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Interface for objects that are able to provide a list of the "first words" associated with a sequence spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20boolean)">fragmentLogProbability(SequenceFragment, boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Computes the total log probability of generating the given sequence fragment under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20boolean)">fragmentLogProbability(SequenceFragment, boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</a></dt>
<dd>
<div class="block">Computes the total log probability of generating the given sequence fragment under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20boolean)">fragmentLogProbability(SequenceFragment, boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Computes the total log probability of generating the given sequence fragment under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#fragmentLogProbability(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20boolean)">fragmentLogProbability(SequenceFragment, boolean)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Computes the total log probability of generating the given sequence fragment under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#fragmentLogProbabilityFromIntKcount(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">fragmentLogProbabilityFromIntKcount(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Computes the total log probability of generating the given sequence fragment under the model.</div>
</dd>
</dl>
<a name="_G_">
<!--   -->
</a>
<h2 class="title">G</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#GAP_BYTE">GAP_BYTE</a></span> - Static variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#gapChars">gapChars</a></span> - Static variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">GapFrequency</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/GapFrequency.html#GapFrequency()">GapFrequency()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/GapFrequency.html" title="class in edu.berkeley.compbio.sequtils">GapFrequency</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/IndelFilter.html#generate()">generate()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils">IndelFilter</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#get(byte[])">get(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given sequence, if it exists</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#get(byte[])">get(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given sequence, if it exists</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAllDownstreamNodes()">getAllDownstreamNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Gets a List<RonPSTNode> of all nodes in the subtree starting from this node, including this node.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getAllUpstreamNodes()">getAllUpstreamNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets a List<RonPSTNode> of all nodes in the subtree starting from this node, including this node.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAlphabet()">getAlphabet()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Returns the alphabet of this SequenceSpectrum object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getAlphabet()">getAlphabet()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Returns the alphabet of this SequenceSpectrum object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getAlphabet()">getAlphabet()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Returns the alphabet of this SequenceSpectrum object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getAncestryList()">getAncestryList()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getArray()">getArray()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>
<div class="block">Returns an array of the counts.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#getArray()">getArray()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>
<div class="block">Returns an array of the counts.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getArraySum()">getArraySum()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getAverageGenomeSize()">getAverageGenomeSize()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getAvgDepth()">getAvgDepth()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getAvgDepth()">getAvgDepth()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getBackoffPrior()">getBackoffPrior()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getBaseSpectrum()">getBaseSpectrum()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Gets the base spectrum from which all other spectra are derived</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getBaseSpectrumRaw()">getBaseSpectrumRaw()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Just get the baseSpectrum as is, with no synchronization or lazy initialization</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#getCharactersRead()">getCharactersRead()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getCharactersRead()">getCharactersRead()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#getCharactersRead()">getCharactersRead()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>
<div class="block">Returns the total number of nucleotides read</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#getCharactersRead()">getCharactersRead()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#getCharactersRead()">getCharactersRead()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getChild(byte)">getChild(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given sequence, if it exists</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getChild(byte)">getChild(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getChildren()">getChildren()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getDescendant(byte[])">getDescendant(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">gets the node associated with the the given sequence, or null if that node does not exist</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getDescendant(byte[])">getDescendant(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getDesiredLength()">getDesiredLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#getEmpty()">getEmpty()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#getEmpty()">getEmpty()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</a></dt>
<dd>
<div class="block">Returns an empty SequenceSpectrum of the appropriate type.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getExclusiveLabel()">getExclusiveLabel()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getExclusiveLabel()">getExclusiveLabel()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getExclusiveLabel()">getExclusiveLabel()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getFactory()">getFactory()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getFirstWordProvider()">getFirstWordProvider()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html#getFirstWords(int)">getFirstWords(int)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/FirstWordProvider.html" title="interface in edu.berkeley.compbio.sequtils.strings">FirstWordProvider</a></dt>
<dd>
<div class="block">Returns a list of all the initial words of length k that are associated with this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getFirstWords(int)">getFirstWords(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getFirstWords(int)">getFirstWords(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getGenomeCount()">getGenomeCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getId()">getId()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Returns a String representing the sequence of symbols leading to this node in the tree.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getId()">getId()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets the id of this node, which is just the suffix associated with this node represented as a byte[], here converted
 to String for the sake of the Clusterable interface.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getId()">getId()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Returns a String identifying this object.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getIdBytes()">getIdBytes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getIdBytes()">getIdBytes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets the id of this node, which is just the suffix associated with this node represented as a byte[].</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getImmutableWeightedLabels()">getImmutableWeightedLabels()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getImmutableWeightedLabels()">getImmutableWeightedLabels()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getItemCount()">getItemCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getItemCount()">getItemCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getK()">getK()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns the pattern length K that this Kcount handles (i.e.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getLength()">getLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns the length of this sequence</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getLength()">getLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Returns the length of this sequence</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#getLevelOneArray()">getLevelOneArray()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getLogProbs()">getLogProbs()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getLongestPrefix(byte[])">getLongestPrefix(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">gets the node associated with the longest available prefix of the given sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#getLongestSuffix(byte[])">getLongestSuffix(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">gets the node associated with the longest available suffix of the given sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getMaxDepth()">getMaxDepth()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Careful-- this gets cached on the first call so will be invalid later if the tree is changed</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getMaxDepth()">getMaxDepth()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getMaxDepth()">getMaxDepth()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Returns the maximum length of substrings considered in computing this statistical model of the sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#getMixingProportion()">getMixingProportion()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getMutableWeightedLabels()">getMutableWeightedLabels()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getMutableWeightedLabels()">getMutableWeightedLabels()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html#getMutuallyExclusiveLabels()">getMutuallyExclusiveLabels()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">LabelProvidingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#getName()">getName()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Returns the species name of this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#getName()">getName()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Returns the species name of this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getName()">getName()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Returns the name of the FastaFileSet</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getName()">getName()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">getName()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Returns the species name of this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getName()">getName()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#getNeedlemanWunsch()">getNeedlemanWunsch()</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow()">getNextWindow()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow(int)">getNextWindow(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow()">getNextWindow()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow(int)">getNextWindow(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow()">getNextWindow()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow(int)">getNextWindow(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow()">getNextWindow()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow(int)">getNextWindow(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#getNucleotideFilter()">getNucleotideFilter()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getNumberOfBins()">getNumberOfBins()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns the number of bins, which is the number of possible patterns in this Kcount.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getNumLeaves()">getNumLeaves()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getNumLeaves()">getNumLeaves()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getNumNodes()">getNumNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getNumNodes()">getNumNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getOriginalSequenceLength()">getOriginalSequenceLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns the length of the sequence that was scanned to produce this spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getOriginalSequenceLength()">getOriginalSequenceLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Returns the number of samples on which this spectrum is based.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getOriginalSequenceLength()">getOriginalSequenceLength()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getOriginalSequenceLength()">getOriginalSequenceLength()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Returns the length of the sequence that was scanned to produce this spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getParent()">getParent()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>
<div class="block">Returns the parent of this Kcount.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getParentMetadata()">getParentMetadata()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns the SequenceFragmentMetadata representing the parent sequence, a longer sequence that contains this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getProbs()">getProbs()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Returns a mapping of symbols to child nodes, describing the possible transitions from this node to its children.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getProbs()">getProbs()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Gets the probability distribution over symbols following the suffix specified by this node's id.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getRawSequence()">getRawSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">getReader(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Returns a SequenceReader at the given section</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20int)">getReader(SequenceFragmentMetadata, int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Returns a SequenceReader at the given section</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getReaderForSynchronizing()">getReaderForSynchronizing()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getReaderForSynchronizing()">getReaderForSynchronizing()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getResetReader()">getResetReader()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getResetReader()">getResetReader()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Returns the number of samples on which the base spectrum is based.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getRootMetadata()">getRootMetadata()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns a String identifying the largest sequence to which this fragment belongs (i.e., the root af the parent
 hierarchy)</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getRootMetadata()">getRootMetadata()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#getRootMetadata()">getRootMetadata()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getRootMetadata()">getRootMetadata()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#getRootMetadata()">getRootMetadata()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getRootSequenceName()">getRootSequenceName()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns a String identifying the largest sequence to which this fragment belongs (i.e., the root af the parent
 hierarchy)</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getScanner()">getScanner()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getSequenceFragmentMetadata(java.lang.String)">getSequenceFragmentMetadata(String)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getSequenceFragments()">getSequenceFragments()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getSequenceName()">getSequenceName()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns a String identifying this sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#getSmithWaterman()">getSmithWaterman()</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getSourceId()">getSourceId()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getSourceId()">getSourceId()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getSpectrum(java.lang.Class,%20com.davidsoergel.dsutils.GenericFactory)">getSpectrum(Class&lt;X&gt;, GenericFactory&lt;X&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Provides a SequenceSpectrum of the requested type.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getStartPosition()">getStartPosition()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns the start position of this sequence, a 0-based index with respect to the parent sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getStartPositionFromRoot()">getStartPositionFromRoot()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getSubtreeNodes()">getSubtreeNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getSumOfCounts()">getSumOfCounts()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns the the sum of the counts.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#getTarget()">getTarget()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#getTaxid()">getTaxid()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Returns the NCBI taxid of the taxon to which this sequence belongs</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getTaxid()">getTaxid()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTaxid()">getTaxid()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Returns the total amount of sequence present in this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Returns the total amount of sequence present in this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#getTotalSequence()">getTotalSequence()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>
<div class="block">Returns the total amount of sequence available</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Returns the total amount of sequence present in all files</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">getTotalSequence()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Returns the total amount of sequence present in this reader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#getTotalSequence()">getTotalSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getTranslatedSequence()">getTranslatedSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#getUnknownCount()">getUnknownCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#getUnknownCount()">getUnknownCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#getUnknownCount()">getUnknownCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getUnknownCount()">getUnknownCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#getUnknownCount()">getUnknownCount()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Returns the number of characters from the original sequence that were not counted because they were unknown</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getUpstreamNode(byte)">getUpstreamNode(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Get the upstream node for the specified symbol.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#getUpstreamNodes()">getUpstreamNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Returns an array of upstream nodes for each symbol, indexed by the alphabet index.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#getWeightedLabels()">getWeightedLabels()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_H_">
<!--   -->
</a>
<h2 class="title">H</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#hasNext()">hasNext()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#hasParent()">hasParent()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#hasParent()">hasParent()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">HierarchicalSpectrum</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="type parameter in HierarchicalSpectrum">T</a> extends <a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#HierarchicalSpectrum()">HierarchicalSpectrum()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_I_">
<!--   -->
</a>
<h2 class="title">I</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#id">id</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#idForSequence(byte[])">idForSequence(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#incrementBy(T)">incrementBy(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">updates this object by adding another one to it.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#incrementBy(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)">incrementBy(MarkovTreeNode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#incrementBy(edu.berkeley.compbio.sequtils.strings.RonPST)">incrementBy(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#incrementBy(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">incrementBy(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode,%20double)">incrementByWeighted(MarkovTreeNode, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.RonPST,%20double)">incrementByWeighted(RonPST, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#incrementByWeighted(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20double)">incrementByWeighted(SequenceFragment, double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#incrementGenomeCount()">incrementGenomeCount()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils"><span class="strong">IndelFilter</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">InferredMixturePST</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span></div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html#InferredMixturePST()">InferredMixturePST()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/InferredMixturePST.html" title="class in edu.berkeley.compbio.sequtils.strings">InferredMixturePST</a></dt>
<dd>
<div class="block"><span class="strong">Deprecated.</span></div>
&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#init()">init()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#init()">init()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#init()">init()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#init()">init()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#init()">init()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initMainBuffer(int)">initMainBuffer(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initTranslationBuffer(int)">initTranslationBuffer(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">IntKcount</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="type parameter in IntKcount">T</a> extends <a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Represents the counts of a set of given patterns appearing in a string.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#IntKcount()">IntKcount()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>
<div class="block">Creates a new Kcount instance.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">IntPseudocountAdder</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#IntPseudocountAdder()">IntPseudocountAdder()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isAligned(byte[][],%20int,%20int,%20int,%20int)">isAligned(byte[][], int, int, int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">Tell whether position pos1 in sequence seq2 is aligned to pos2 in seq2 in the given char array.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isGap(byte)">isGap(byte)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#isIgnoreEdges()">isIgnoreEdges()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#isLeaf()">isLeaf()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#isLeaf()">isLeaf()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Tells whether this node is a leaf or not.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#isLeftJustified(byte[])">isLeftJustified(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#isMatePair(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">isMatePair(SequenceFragmentMetadata, SequenceFragmentMetadata)</a></span> - Static method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#iupacPattern(java.lang.String)">iupacPattern(String)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_J_">
<!--   -->
</a>
<h2 class="title">J</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#joinMatePair(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20edu.berkeley.compbio.sequtils.strings.SequenceFragment)">joinMatePair(SequenceFragment, SequenceFragment)</a></span> - Static method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_K_">
<!--   -->
</a>
<h2 class="title">K</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#k">k</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">Kcount</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="type parameter in Kcount">T</a> extends <a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#Kcount()">Kcount()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings"><span class="strong">KcountScanner</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Provides an interface for classes that can scan sequences to count the occurrences of substrings</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">KneserNeyPSASmoother</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#KneserNeyPSASmoother()">KneserNeyPSASmoother()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_L_">
<!--   -->
</a>
<h2 class="title">L</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#label">label</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelProvidingSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">LabelProvidingSFI</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">LabelSelectableSFI</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">LairdSaulPST</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html#LairdSaulPST(java.io.Reader)">LairdSaulPST(Reader)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/LairdSaulPST.html" title="class in edu.berkeley.compbio.sequtils.strings">LairdSaulPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#lastSymbolForId(int)">lastSymbolForId(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#leaf">leaf</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#learn(double,%20double,%20double,%20int,%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrum,%20com.davidsoergel.stats.DistributionProcessor)">learn(double, double, double, int, SequenceSpectrum, DistributionProcessor&lt;RonPSA&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#length">length</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">The total number of characters represented by this fragment, including "N" characters or whatever.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#logConditionalProbability(byte)">logConditionalProbability(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Returns the conditional probability of the given symbol from this node.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#logConditionalProbabilityByAlphabetIndex(int)">logConditionalProbabilityByAlphabetIndex(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">LogOddsSubstitutionMatrix</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>
<div class="block">A substitution matrix with integer scores (in log probability units), such as a PAM or BLOSUM matrix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#LogOddsSubstitutionMatrix()">LogOddsSubstitutionMatrix()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_M_">
<!--   -->
</a>
<h2 class="title">M</h2>
<dl>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">MarkovTreeNode</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">A node in a tree where each transition is labeled by a byte and has a local probability (that is, conditional on the
 parent node).</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#MarkovTreeNode()">MarkovTreeNode()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#MarkovTreeNode(byte[],%20byte[])">MarkovTreeNode(byte[], byte[])</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Constructs a new MarkovTreeNode with the given identifier</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#minus(T)">minus(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns a new object representing the difference between this one and the given argument.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#minus(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)">minus(MarkovTreeNode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#minus(edu.berkeley.compbio.sequtils.strings.RonPST)">minus(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#minus(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">minus(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#multiplyBy(double)">multiplyBy(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#multiplyBy(double)">multiplyBy(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#multiplyBy(double)">multiplyBy(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#multiplyBy(double)">multiplyBy(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_N_">
<!--   -->
</a>
<h2 class="title">N</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#newParent()">newParent()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>
<div class="block">Generates a new Kcount based on this one and stores it as the parent</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Returns the next section</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#next()">next()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()">next()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">Iterates through the list of sections in order.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#nextFullyLabelled()">nextFullyLabelled()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNode(byte)">nextNode(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">Gets the child node associated with the given sequence, if it exists</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNodeByAlphabetIndex(int)">nextNodeByAlphabetIndex(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#nextNodes()">nextNodes()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html#nextRequiringLabel(java.lang.String)">nextRequiringLabel(String)</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/LabelSelectableSFI.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">LabelSelectableSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()">nextShuffled()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#nextShuffled()">nextShuffled()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#nextShuffled()">nextShuffled()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">nextShuffled()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">Iterates through the list of sections in random order, providing each section exactly once</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#normalize()">normalize()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Normalize the underlying probability distribution</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">NotEnoughSequenceException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>
<div class="block">Used when there is insufficient sequence in a reader for the next operation, such as the end of section encountered,
 or being inside a header.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.String)">NotEnoughSequenceException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.Exception)">NotEnoughSequenceException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html#NotEnoughSequenceException(java.lang.Exception,%20java.lang.String)">NotEnoughSequenceException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#numberOfBins">numberOfBins</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/IndelFilter.html#numChars()">numChars()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/IndelFilter.html" title="interface in edu.berkeley.compbio.sequtils">IndelFilter</a></dt>
<dd>
<div class="block">return the number of nucleotides to add or remove at a given nucletodie position.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numChars(byte[])">numChars(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGapClusters(byte[])">numGapClusters(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGaps(byte[])">numGaps(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numGaps(char[])">numGaps(char[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#numNonGaps(byte[])">numNonGaps(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_O_">
<!--   -->
</a>
<h2 class="title">O</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#originalSequenceLength">originalSequenceLength</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#overlaps(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">overlaps(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Tells whether this fragment overlaps with another.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#overlaps(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">overlaps(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_P_">
<!--   -->
</a>
<h2 class="title">P</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#padEndWithGaps(byte[],%20int)">padEndWithGaps(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#padStartWithGaps(byte[],%20int)">padStartWithGaps(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">PairAlignedDnaDist</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#PairAlignedDnaDist(edu.berkeley.compbio.sequtils.DnaDist,%20edu.berkeley.compbio.sequtils.DynamicProgrammingPairwiseAligner)">PairAlignedDnaDist(DnaDist, DynamicProgrammingPairwiseAligner)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#parent">parent</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#parentMetadata">parentMetadata</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#peek()">peek()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#plus(T)">plus(T)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>
<div class="block">Returns a new object representing the sum of this one and the given argument.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#plus(edu.berkeley.compbio.sequtils.strings.MarkovTreeNode)">plus(MarkovTreeNode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#plus(edu.berkeley.compbio.sequtils.strings.RonPST)">plus(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#plus(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">plus(SequenceFragment)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#prefixForId(int)">prefixForId(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#prefixId(int)">prefixId(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#probs">probs</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#process(com.davidsoergel.stats.DoubleArrayContainer)">process(DoubleArrayContainer)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#process(com.davidsoergel.stats.IntArrayContainer)">process(IntArrayContainer)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPSA)">process(RonPSA)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPSA)">process(RonPSA)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#process(edu.berkeley.compbio.sequtils.strings.RonPST)">process(RonPST)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_R_">
<!--   -->
</a>
<h2 class="title">R</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()">randomSectionFragment()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionFragment()">randomSectionFragment()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionFragment()">randomSectionFragment()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">randomSectionFragment()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionLengthWeighted()">randomSectionLengthWeighted()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">Get a random section from the list, selected uniformly from the complete sections in the list, weighted by their
 lengths.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader"><span class="strong">RandomSectionList</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a></dt>
<dd>
<div class="block">Provides randomly generated sequence implementing SequenceReader</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#RandomSectionList()">RandomSectionList()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">RandomSectionSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads sections.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#RandomSectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">RandomSectionSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionUniform()">randomSectionUniform()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionUniform()">randomSectionUniform()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionUniform()">randomSectionUniform()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()">randomSectionUniform()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">RandomSequentialMatePairSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads random mate-pairs</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#RandomSequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">RandomSequentialMatePairSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#RandomSequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList,%20int)">RandomSequentialMatePairSFI(SectionList, int)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read()">read()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read(byte[],%20int)">read(byte[], int)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Reads nucleotides into a byte array</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read()">read()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read(byte[],%20int)">read(byte[], int)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Reads nucleotides into a byte array</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()">read()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read(byte[],%20int)">read(byte[], int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read()">read()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read(byte[],%20int)">read(byte[], int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#read(int)">read(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[],%20int)">read(byte[], int)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Reads nucleotides into a byte array</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#readAt(byte[],%20long,%20int)">readAt(byte[], long, int)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Reads a section from the files</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#readTranslated()">readTranslated()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#readTranslated()">readTranslated()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()">readTranslated()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html#realign(com.davidsoergel.dsutils.collections.OrderedPair)">realign(OrderedPair&lt;byte[], byte[]&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.html" title="class in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#releaseCachedResources()">releaseCachedResources()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Closes all the open filehandles</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#remove()">remove()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</a></dt>
<dd>
<div class="block">Unsupported, as all iteration is online</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#remove()">remove()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#removeOverlaps(java.util.Collection)">removeOverlaps(Collection&lt;SequenceFragment&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Careful: this does not propagate labels to the subsidiary fragments, because they may or may not apply.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#replaceGaps(byte[])">replaceGaps(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanRawSequence()">rescanRawSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#rescanSpectrum()">rescanSpectrum()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">AggregateSequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanSpectrum()">rescanSpectrum()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanTranslatedSequence()">rescanTranslatedSequence()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">reset the stream to the beginning.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">resets the base reader.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">RandomSequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#reset()">reset()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#reset()">reset()</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">reset the stream to the beginning.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()">reset()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dt>
<dd>
<div class="block">reset the stream to the beginning.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#reverseComplement(byte[])">reverseComplement(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPSA</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html#RonPSA()">RonPSA()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSA.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSA</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPSANode</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Extends a MarkovTreeNode with backlinks, for making make a Probabilistic Suffix Automaton (PSA).</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#RonPSANode()">RonPSANode()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#RonPSANode(byte[],%20byte[])">RonPSANode(byte[], byte[])</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">Constructs a new MarkovTreeNode with the given identifier</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPSASmoother</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#RonPSASmoother()">RonPSASmoother()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPST</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">A probabilistic suffix tree, as described in
 <p/>
 D.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#RonPST()">RonPST()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#RonPST(double,%20double,%20double,%20int,%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">RonPST(double, double, double, int, SequenceSpectrum)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Constructs a new Probabilistic Suffix Tree according to the Learn-PST algorithm (Ron et al 1996 p.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPSTNode</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">A node of a Probabilistic Suffix Tree, where each symbol is a byte.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#RonPSTNode()">RonPSTNode()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#RonPSTNode(byte[],%20byte[])">RonPSTNode(byte[], byte[])</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Constructs a new RonPSTNode with the given identifier</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">RonPSTSmoother</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#RonPSTSmoother()">RonPSTSmoother()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotate(byte[],%20int,%20int)">rotate(byte[], int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotate(byte[],%20int,%20int,%20int)">rotate(byte[], int, int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">Rotates an array slice in place within the larger array.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotateUpdatingGaps(int[],%20int,%20int)">rotateUpdatingGaps(int[], int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#rotateUpdatingGaps(int[],%20int,%20int,%20int)">rotateUpdatingGaps(int[], int, int, int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#runPhaseInit()">runPhaseInit()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#runPhaseInit()">runPhaseInit()</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_S_">
<!--   -->
</a>
<h2 class="title">S</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#sample(byte[])">sample(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#sample(int)">sample(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Chooses a random string according to the conditional probabilities of symbols.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#sample(byte[])">sample(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#sample(int)">sample(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Chooses a random string according to the conditional probabilities of symbols.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#sample(byte[])">sample(byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Chooses a random symbol according to the conditional probabilities of symbols following the given prefix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#sample(int)">sample(int)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Chooses a random string according to the conditional probabilities of symbols.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">ScanningSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#ScanningSFI()">ScanningSFI()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#ScanningSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">ScanningSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence(SequenceFragment)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">KcountScanner</a></dt>
<dd>
<div class="block">Scans a sequence to count pattern frequencies.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20byte[])">scanSequence(SequenceFragment, byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KcountScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">KcountScanner</a></dt>
<dd>
<div class="block">Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence(SequenceFragment)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</a></dt>
<dd>
<div class="block">Scans a sequence to count pattern frequencies.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20byte[])">scanSequence(SequenceFragment, byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumScanner</a></dt>
<dd>
<div class="block">Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html#score(byte,%20byte,%20boolean)">score(byte, byte, boolean)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/AffineSubstitutionMatrix.html" title="interface in edu.berkeley.compbio.sequtils">AffineSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#score(byte,%20byte)">score(byte, byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html#score(byte,%20byte,%20boolean)">score(byte, byte, boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">SimpleSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html#scoreAgainstGap(byte)">scoreAgainstGap(byte)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/LogOddsSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">LogOddsSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">SectionDecomposingSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads section lists</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#SectionDecomposingSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">SectionDecomposingSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionDecomposingSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#sectionList">sectionList</a></span> - Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader"><span class="strong">SectionList</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a></dt>
<dd>
<div class="block">A list of sequence fragments, represented only by their metadata.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">SectionListBasedSequenceFragmentIterator</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#SectionListBasedSequenceFragmentIterator(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">SectionListBasedSequenceFragmentIterator(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html#SectionListBasedSequenceFragmentIterator()">SectionListBasedSequenceFragmentIterator()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionListBasedSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionListBasedSequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">SectionSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads sections.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#SectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">SectionSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(long)">seek(long)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek(SequenceFragmentMetadata)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek(SequenceFragmentMetadata, long)</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#seqIndexToColumn(byte[],%20int)">seqIndexToColumn(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">Return the column (0-based) in which the n'th (0-based) non-gap character appears.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceArrayException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.String)">SequenceArrayException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.Exception)">SequenceArrayException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html#SequenceArrayException(java.lang.Exception,%20java.lang.String)">SequenceArrayException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayException.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceArrayUtils</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#SequenceArrayUtils()">SequenceArrayUtils()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceError</span></a> - Error in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.String)">SequenceError(String)</a></span> - Constructor for error edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.Exception)">SequenceError(Exception)</a></span> - Constructor for error edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceError.html#SequenceError(java.lang.Exception,%20java.lang.String)">SequenceError(Exception, String)</a></span> - Constructor for error edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceError.html" title="class in edu.berkeley.compbio.sequtils">SequenceError</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.String)">SequenceException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.Exception)">SequenceException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceException.html#SequenceException(java.lang.Exception,%20java.lang.String)">SequenceException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#sequenceForId(int)">sequenceForId(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">SequenceFragment</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Manages information about a sequence fragment, including its metadata and any statistics that have been calculated
 from the sequence, in the form of SequenceSpectra.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20long,%20long,%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)">SequenceFragment(SequenceFragmentMetadata, String, long, long, SequenceSpectrumScanner)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Constructs a new SequenceFragment by specifying its coordinates with respect to a containing parent sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20long,%20edu.berkeley.compbio.sequtils.SequenceReader,%20long,%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)">SequenceFragment(SequenceFragmentMetadata, String, long, SequenceReader, long, SequenceSpectrumScanner)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20long,%20edu.berkeley.compbio.sequtils.SequenceReader,%20long,%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner,%20long)">SequenceFragment(SequenceFragmentMetadata, String, long, SequenceReader, long, SequenceSpectrumScanner, long)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Convenience constructor, creates a new SequenceFragment by applying a SequenceSpectrumScanner to an input stream,
 and using the resulting SequenceSpectrum as the base spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#SequenceFragment(byte[],%20edu.berkeley.compbio.sequtils.strings.SequenceSpectrumScanner)">SequenceFragment(byte[], SequenceSpectrumScanner)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">SequenceFragmentIterator</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceFragmentMetadata</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="type parameter in SequenceFragmentMetadata">C</a> extends <a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>
<div class="block">Provides a container for information about a particular sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#SequenceFragmentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20java.lang.Integer,%20int)">SequenceFragmentMetadata(SequenceFragmentMetadata, String, Integer, int)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Constructs a new SequenceFragmentMetadata by specifying its start position with respect to a containing parent
 sequence, but with an unknown length.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#SequenceFragmentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20java.lang.String,%20java.lang.Integer,%20long,%20long)">SequenceFragmentMetadata(SequenceFragmentMetadata, String, Integer, long, long)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Constructs a new SequenceFragmentMetadata by specifying its coordinates with respect to a containing parent
 sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#sequenceName">sequenceName</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils"><span class="strong">SequenceReader</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>
<div class="block">Provides an interface for a buffered sequence reader, to return nucleotides and process parsers or mutation
 transparently.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SequenceRuntimeException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.String)">SequenceRuntimeException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.Exception)">SequenceRuntimeException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html#SequenceRuntimeException(java.lang.Exception,%20java.lang.String)">SequenceRuntimeException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils">SequenceRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings"><span class="strong">SequenceSpectrum</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="type parameter in SequenceSpectrum">T</a> extends <a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a>&gt; - Interface in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Represents a statistical model of a sequence of bytes.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">SequenceSpectrumException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Thrown when something involving a SequenceSpectrum goes wrong, such as when a requested spectrum cannot be computed.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.String)">SequenceSpectrumException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.Exception)">SequenceSpectrumException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html#SequenceSpectrumException(java.lang.Exception,%20java.lang.String)">SequenceSpectrumException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">SequenceSpectrumRuntimeException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Thrown when something involving a SequenceSpectrum goes wrong, such as when a requested spectrum cannot be computed.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.String)">SequenceSpectrumRuntimeException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.Exception)">SequenceSpectrumRuntimeException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html#SequenceSpectrumRuntimeException(java.lang.Exception,%20java.lang.String)">SequenceSpectrumRuntimeException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumRuntimeException.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceSpectrumRuntimeException</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html" title="interface in edu.berkeley.compbio.sequtils.strings"><span class="strong">SequenceSpectrumScanner</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>
<div class="block">Provides an interface for classes that can scan sequences to count the occurrences of substrings</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">SequentialMatePairSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads sequential mate-pairs</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html#SequentialMatePairSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">SequentialMatePairSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequentialMatePairSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequentialMatePairSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setAlphabet(byte[])">setAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setAlphabet(byte[])">setAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#setAlphabet(byte[])">setAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Sets the array of valid symbols.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setBacklinksUsingRoot(edu.berkeley.compbio.sequtils.strings.RonPSANode,%20java.util.Queue)">setBacklinksUsingRoot(RonPSANode, Queue&lt;RonPSANode&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html#setBackoffPrior(edu.berkeley.compbio.sequtils.strings.RonPSANode)">setBackoffPrior(RonPSANode)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSANode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSANode</a></dt>
<dd>
<div class="block">This method assumes that it has not been run before, and that all nodes at higher levels in the tree have already
 been proceesed.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setBaseSpectrum(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">setBaseSpectrum(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Sets the base spectrum from which all other spectra are derived</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">setFragmentLength(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>
<div class="block">Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setId(byte[])">setId(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#setId(byte[])">setId(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Sets the id of this node, which is just the suffix associated with this node represented as a byte[].</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setIgnoreEdges(boolean)">setIgnoreEdges(boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setIgnoreEdges(boolean)">setIgnoreEdges(boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Specify whether or not this sequence spectrum attempts to account for edge effects, i.e.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setIgnoreEdges(boolean)">setIgnoreEdges(boolean)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#setIgnoreEdges(boolean)">setIgnoreEdges(boolean)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Specify whether or not this sequence spectrum attempts to account for edge effects, i.e.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setImmutable()">setImmutable()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setImmutable()">setImmutable()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Declare that this spectrum should not change anymore.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#setImmutable()">setImmutable()</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Declare that this spectrum should not change anymore.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html#setLabel(java.lang.String)">setLabel(String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/HierarchicalSpectrum.html" title="class in edu.berkeley.compbio.sequtils.strings">HierarchicalSpectrum</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setLabel(java.lang.String)">setLabel(String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#setLabel(java.lang.String)">setLabel(String)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setLength(int)">setLength(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Sets the length of this sequence</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setOriginalSequenceLength(long)">setOriginalSequenceLength(long)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setParentMetadata(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">setParentMetadata(SequenceFragmentMetadata)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Sets the SequenceFragmentMetadata representing the parent sequence, a longer sequence that contains this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html#setResults(com.davidsoergel.trees.dhtpn.SerializableDoubleHierarchicalTypedProperties)">setResults(SerializableDoubleHierarchicalTypedProperties&lt;?&gt;)</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/AbstractSequenceFragmentIterator.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">AbstractSequenceFragmentIterator</a></dt>
<dd>
<div class="block">Frees up memory, filehandles, etc.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html#setResults(com.davidsoergel.trees.dhtpn.SerializableDoubleHierarchicalTypedProperties)">setResults(SerializableDoubleHierarchicalTypedProperties&lt;?&gt;)</a></span> - Method in interface edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SequenceFragmentIterator.html" title="interface in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SequenceFragmentIterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setSequenceName(java.lang.String)">setSequenceName(String)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Sets a String identifying this sequence</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#setStartPosition(long)">setStartPosition(long)</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>
<div class="block">Sets the start position of this sequence, a 0-based index with respect to the parent sequence.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a></dt>
<dd>
<div class="block">Set the character mapping to be used when reading nucleotides.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#setTranslationAlphabet(byte[])">setTranslationAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.<a href="./edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet(byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></dt>
<dd>
<div class="block">Set the character mapping to be used when reading nucleotides.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#setTranslationAlphabet(byte[])">setTranslationAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>
<div class="block">Set the character mapping to be used when reading nucleotides.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationInvalid()">setTranslationInvalid()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#setUnknownCount(long)">setUnknownCount(long)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">ShuffledSectionSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>
<div class="block">Provides a SequenceFragmentIterator that reads sections.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#ShuffledSectionSFI(edu.berkeley.compbio.sequtils.sequencereader.SectionList)">ShuffledSectionSFI(SectionList)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ShuffledSectionSFI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">SimpleSubstitutionMatrix</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html#SimpleSubstitutionMatrix(float,%20float,%20float,%20float)">SimpleSubstitutionMatrix(float, float, float, float)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SimpleSubstitutionMatrix.html" title="class in edu.berkeley.compbio.sequtils">SimpleSubstitutionMatrix</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings"><span class="strong">SimplexVectorTargetAndProportion</span></a>&lt;<a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="type parameter in SimplexVectorTargetAndProportion">T</a>&gt; - Class in <a href="./edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html#SimplexVectorTargetAndProportion(com.davidsoergel.stats.Multinomial,%20com.davidsoergel.stats.Multinomial)">SimplexVectorTargetAndProportion(Multinomial&lt;T&gt;, Multinomial&lt;T&gt;)</a></span> - Constructor for class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SimplexVectorTargetAndProportion.html" title="class in edu.berkeley.compbio.sequtils.strings">SimplexVectorTargetAndProportion</a></dt>
<dd>
<div class="block">Constructs a new RelativeMarkovTreeNode by comparing one MarkovTreeNode with another and representing the difference
 as a vector on the simplex.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html#smoothFactor">smoothFactor</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/KneserNeyPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">KneserNeyPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html#smoothFactor">smoothFactor</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSASmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSASmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html#smoothFactor">smoothFactor</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTSmoother.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTSmoother</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoubleKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">DoubleKcount</a></dt>
<dd>
<div class="block">Tests whether the given sequence statistics are equivalent to this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntKcount.html" title="class in edu.berkeley.compbio.sequtils.strings">IntKcount</a></dt>
<dd>
<div class="block">Test whether the given sequence statistics are equivalent to this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Test whether the given sequence statistics are equivalent to this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Test whether the given sequence statistics are equivalent to this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>
<div class="block">Test whether this sequence has statistics matching the given spectrum.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#spectrumEquals(edu.berkeley.compbio.sequtils.strings.SequenceSpectrum)">spectrumEquals(SequenceSpectrum)</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Test whether the given sequence statistics are equivalent to this one.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html#spectrumScanner">spectrumScanner</a></span> - Variable in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/ScanningSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">ScanningSFI</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#startPosition">startPosition</a></span> - Variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startPositionOfGapCluster(byte[],%20int)">startPositionOfGapCluster(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithGaps(byte[],%20int)">startsWithGaps(byte[], int)</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithIUPAC(byte[],%20byte[])">startsWithIUPAC(byte[], byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#startsWithIUPAC(byte[],%20int,%20byte[])">startsWithIUPAC(byte[], int, byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#stripGaps(byte[][])">stripGaps(byte[][])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">Strip all gaps, left-justifying the sequences.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#stripGaps(byte[])">stripGaps(byte[])</a></span> - Static method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>
<div class="block">Push all the gaps to the end of the array, in place</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator"><span class="strong">StubSFI</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html#StubSFI()">StubSFI()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/StubSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">StubSFI</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/SubstitutionFilter.html" title="interface in edu.berkeley.compbio.sequtils"><span class="strong">SubstitutionFilter</span></a> - Interface in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>
<div class="block">A filter that mutates nucleotides one at a time (i.e., without taking context into account).</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#suffixId(int)">suffixId(int)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_T_">
<!--   -->
</a>
<h2 class="title">T</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#theDerivedSpectra">theDerivedSpectra</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#theScanner">theScanner</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#times(double)">times(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#times(double)">times(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#times(double)">times(double)</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#toLongString()">toLongString()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#toLongString()">toLongString()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DnaDist.html#toString()">toString()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DnaDist.html" title="class in edu.berkeley.compbio.sequtils">DnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html#toString()">toString()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/PairAlignedDnaDist.html" title="class in edu.berkeley.compbio.sequtils">PairAlignedDnaDist</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#toString()">toString()</a></span> - Method in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#toString()">toString()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a></dt>
<dd>
<div class="block">Returns the name of the FastaFileSet</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#toString()">toString()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencereader.fasta.<a href="./edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#totalFragment()">totalFragment()</a></span> - Method in class edu.berkeley.compbio.sequtils.sequencefragmentiterator.<a href="./edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html" title="class in edu.berkeley.compbio.sequtils.sequencefragmentiterator">SectionSFI</a></dt>
<dd>
<div class="block">Note this is partly redundant with RepeatingAggregatingSFI</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#totalProbability(byte[])">totalProbability(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>
<div class="block">Computes the probability of generating the given sequence under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html#totalProbability(byte[])">totalProbability(byte[])</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPST.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPST</a></dt>
<dd>
<div class="block">Computes the probability of generating the given sequence under the model.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html#totalProbability(byte[])">totalProbability(byte[])</a></span> - Method in interface edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></dt>
<dd>
<div class="block">Computes the probability of generating the given sequence under the model.</div>
</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">TranslatingSequenceReader</span></a> - Class in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#TranslatingSequenceReader()">TranslatingSequenceReader()</a></span> - Constructor for class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dt>
<dd>&nbsp;</dd>
<dt><a href="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">TranslationException</span></a> - Exception in <a href="./edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.String)">TranslationException(String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.Exception)">TranslationException(Exception)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/TranslationException.html#TranslationException(java.lang.Exception,%20java.lang.String)">TranslationException(Exception, String)</a></span> - Constructor for exception edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</a></dt>
<dd>&nbsp;</dd>
</dl>
<a name="_U_">
<!--   -->
</a>
<h2 class="title">U</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html#uniformPseudoCount">uniformPseudoCount</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/DoublePseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">DoublePseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html#uniformPseudoCount">uniformPseudoCount</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/IntPseudocountAdder.html" title="class in edu.berkeley.compbio.sequtils.strings">IntPseudocountAdder</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#UNKNOWN_BYTE">UNKNOWN_BYTE</a></span> - Static variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceArrayUtils.html" title="class in edu.berkeley.compbio.sequtils">SequenceArrayUtils</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html#UNKNOWN_LENGTH">UNKNOWN_LENGTH</a></span> - Static variable in class edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#UNKNOWN_NUMBER_OF_SAMPLES">UNKNOWN_NUMBER_OF_SAMPLES</a></span> - Static variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#UNKNOWN_SUM_OF_COUNTS">UNKNOWN_SUM_OF_COUNTS</a></span> - Static variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#unknownCount">unknownCount</a></span> - Variable in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html#unknownProportion()">unknownProportion()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#updateLogProbs()">updateLogProbs()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#updateLogProbs()">updateLogProbs()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Takes the logs of all the probabilities at this node and caches them.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html#updateLogProbsRecursive()">updateLogProbsRecursive()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/MarkovTreeNode.html" title="class in edu.berkeley.compbio.sequtils.strings">MarkovTreeNode</a></dt>
<dd>&nbsp;</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html#updateLogProbsRecursive()">updateLogProbsRecursive()</a></span> - Method in class edu.berkeley.compbio.sequtils.strings.<a href="./edu/berkeley/compbio/sequtils/strings/RonPSTNode.html" title="class in edu.berkeley.compbio.sequtils.strings">RonPSTNode</a></dt>
<dd>
<div class="block">Takes the logs of all the probabilities at this node and caches them; then follows the upstream transitions and
 repeats recursively.</div>
</dd>
</dl>
<a name="_V_">
<!--   -->
</a>
<h2 class="title">V</h2>
<dl>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html#valueOf(java.lang.String)">valueOf(String)</a></span> - Static method in enum edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackBegin</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html#valueOf(java.lang.String)">valueOf(String)</a></span> - Static method in enum edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackEnd</a></dt>
<dd>
<div class="block">Returns the enum constant of this type with the specified name.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html#values()">values()</a></span> - Static method in enum edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackBegin.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackBegin</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
<dt><span class="strong"><a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html#values()">values()</a></span> - Static method in enum edu.berkeley.compbio.sequtils.<a href="./edu/berkeley/compbio/sequtils/DynamicProgrammingPairwiseAligner.TracebackEnd.html" title="enum in edu.berkeley.compbio.sequtils">DynamicProgrammingPairwiseAligner.TracebackEnd</a></dt>
<dd>
<div class="block">Returns an array containing the constants of this enum type, in
the order they are declared.</div>
</dd>
</dl>
<a href="#_A_">A</a>&nbsp;<a href="#_B_">B</a>&nbsp;<a href="#_C_">C</a>&nbsp;<a href="#_D_">D</a>&nbsp;<a href="#_E_">E</a>&nbsp;<a href="#_F_">F</a>&nbsp;<a href="#_G_">G</a>&nbsp;<a href="#_H_">H</a>&nbsp;<a href="#_I_">I</a>&nbsp;<a href="#_J_">J</a>&nbsp;<a href="#_K_">K</a>&nbsp;<a href="#_L_">L</a>&nbsp;<a href="#_M_">M</a>&nbsp;<a href="#_N_">N</a>&nbsp;<a href="#_O_">O</a>&nbsp;<a href="#_P_">P</a>&nbsp;<a href="#_R_">R</a>&nbsp;<a href="#_S_">S</a>&nbsp;<a href="#_T_">T</a>&nbsp;<a href="#_U_">U</a>&nbsp;<a href="#_V_">V</a>&nbsp;</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="./overview-summary.html">Overview</a></li>
<li>Package</li>
<li>Class</li>
<li>Use</li>
<li><a href="./overview-tree.html">Tree</a></li>
<li><a href="./deprecated-list.html">Deprecated</a></li>
<li class="navBarCell1Rev">Index</li>
<li><a href="./help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="./index.html?index-all.html" target="_top">Frames</a></li>
<li><a href="index-all.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="./allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.</small></p>
</body>
</html>
