<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_60) on Tue Jul 08 13:00:11 EDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>SequenceReader (sequtils 0.92-SNAPSHOT API)</title>
<meta name="date" content="2014-07-08">
<link rel="stylesheet" type="text/css" href="../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="SequenceReader (sequtils 0.92-SNAPSHOT API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceReader.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?edu/berkeley/compbio/sequtils/SequenceReader.html" target="_top">Frames</a></li>
<li><a href="SequenceReader.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">edu.berkeley.compbio.sequtils</div>
<h2 title="Interface SequenceReader" class="title">Interface SequenceReader</h2>
</div>
<div class="contentContainer">
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Known Subinterfaces:</dt>
<dd><a href="../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dd>
</dl>
<dl>
<dt>All Known Implementing Classes:</dt>
<dd><a href="../../../../edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html" title="class in edu.berkeley.compbio.sequtils">ByteArraySequenceReader</a>, <a href="../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaParser</a>, <a href="../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">FilteringSectionList</a>, <a href="../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">FilteringSequenceReader</a>, <a href="../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html" title="class in edu.berkeley.compbio.sequtils.sequencereader">RandomSectionList</a>, <a href="../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></dd>
</dl>
<hr>
<br>
<pre>public interface <span class="strong">SequenceReader</span></pre>
<div class="block">Provides an interface for a buffered sequence reader, to return nucleotides and process parsers or mutation
 transparently.</div>
<dl><dt><span class="strong">Version:</span></dt>
  <dd>$Id$</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>David Tulga</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#close()">close</a></strong>()</code>
<div class="block">Closes the reader, and all sub-readers</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">getName</a></strong>()</code>
<div class="block">Returns the species name of this reader</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">getTotalSequence</a></strong>()</code>
<div class="block">Returns the total amount of sequence present in this reader</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read</a></strong>()</code>
<div class="block">Returns the next nucleotide</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[],%20int)">read</a></strong>(byte[]&nbsp;buffer,
    int&nbsp;length)</code>
<div class="block">Reads nucleotides into a byte array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated</a></strong>()</code>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#reset()">reset</a></strong>()</code>
<div class="block">reset the stream to the beginning.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek</a></strong>(<a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section)</code>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek</a></strong>(<a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section,
    long&nbsp;offset)</code>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>boolean</code></td>
<td class="colLast"><code><strong><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet</a></strong>(byte[]&nbsp;alphabet)</code>
<div class="block">Set the character mapping to be used when reading nucleotides.</div>
</td>
</tr>
</table>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="close()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>close</h4>
<pre>void&nbsp;close()</pre>
<div class="block">Closes the reader, and all sub-readers</div>
</li>
</ul>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block">Returns the species name of this reader</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>A String representing the name of the sequence</dd></dl>
</li>
</ul>
<a name="getTotalSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalSequence</h4>
<pre>long&nbsp;getTotalSequence()</pre>
<div class="block">Returns the total amount of sequence present in this reader</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The total amount of sequence</dd></dl>
</li>
</ul>
<a name="read()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>byte&nbsp;read()
          throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                 <a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a>,
                 <a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></pre>
<div class="block">Returns the next nucleotide</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The next nucleotide, or EOF if the section or file has ended</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="read(byte[], int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>int&nbsp;read(byte[]&nbsp;buffer,
       int&nbsp;length)
         throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                <a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a>,
                <a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></pre>
<div class="block">Reads nucleotides into a byte array</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>buffer</code> - a byte[] to read into.</dd><dd><code>length</code> - how many nucleotides to read (must be less than or equal to the size of the given buffer)</dd>
<dt><span class="strong">Returns:</span></dt><dd>The number of nucleotides read</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="readTranslated()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>readTranslated</h4>
<pre>int&nbsp;readTranslated()
                   throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                          <a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a>,
                          <a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a>,
                          <a href="../../../../edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</a></pre>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.
 Typically used to map Roman characters representing nucleotides to the integers 0-3.</div>
<dl><dt><span class="strong">Returns:</span></dt><dd>The next nucleotide, using the translated alphabet, or EOF if the section or file has ended</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/FilterException.html" title="class in edu.berkeley.compbio.sequtils">FilterException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd>
<dd><code><a href="../../../../edu/berkeley/compbio/sequtils/TranslationException.html" title="class in edu.berkeley.compbio.sequtils">TranslationException</a></code></dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])"><code>setTranslationAlphabet(byte[] alphabet)</code></a></dd></dl>
</li>
</ul>
<a name="reset()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reset</h4>
<pre>void&nbsp;reset()</pre>
<div class="block">reset the stream to the beginning.</div>
</li>
</ul>
<a name="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seek</h4>
<pre>void&nbsp;seek(<a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section)
          throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>section</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seek</h4>
<pre>void&nbsp;seek(<a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section,
        long&nbsp;offset)
          throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>section</code> - </dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="setTranslationAlphabet(byte[])">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>setTranslationAlphabet</h4>
<pre>boolean&nbsp;setTranslationAlphabet(byte[]&nbsp;alphabet)</pre>
<div class="block">Set the character mapping to be used when reading nucleotides.</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>alphabet</code> - A byte[] representing the byte translation table.  The mapping is specified in reverse: the desired
                 translated value is the index of the input array, and the values in the array are the corresponding
                 input characters to be mapped.  For example, the mapping 'A'->0, 'C'->1, 'G'->2, 'T'->3 is
                 represented as new byte[]{'A','C','G','T'}.</dd><dt><span class="strong">See Also:</span></dt><dd><a href="../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()"><code>readTranslated()</code></a></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/SequenceReader.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../index-all.html">Index</a></li>
<li><a href="../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../edu/berkeley/compbio/sequtils/SequenceRuntimeException.html" title="class in edu.berkeley.compbio.sequtils"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../index.html?edu/berkeley/compbio/sequtils/SequenceReader.html" target="_top">Frames</a></li>
<li><a href="SequenceReader.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li>Constr&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.</small></p>
</body>
</html>
