<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Sun Jul 06 18:05:53 EDT 2014 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
FastaParser (sequtils 0.92-SNAPSHOT API)
</TITLE>

<META NAME="date" CONTENT="2014-07-06">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastaParser (sequtils 0.92-SNAPSHOT API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastaParser.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastaParser.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<!-- ======== START OF CLASS DATA ======== -->
<H2>
<FONT SIZE="-1">
edu.berkeley.compbio.sequtils.sequencereader.fasta</FONT>
<BR>
Class FastaParser</H2>
<PRE>
<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</A>
  <IMG SRC="../../../../../../resources/inherit.gif" ALT="extended by "><A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">edu.berkeley.compbio.sequtils.TranslatingSequenceReader</A>
      <IMG SRC="../../../../../../resources/inherit.gif" ALT="extended by "><B>edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaParser</B>
</PRE>
<DL>
<DT><B>All Implemented Interfaces:</B> <DD><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></DD>
</DL>
<HR>
<DL>
<DT><PRE>public class <B>FastaParser</B><DT>extends <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A><DT>implements <A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></DL>
</PRE>

<P>
Fasta File Parser
<P>

<P>
<DL>
<DT><B>Version:</B></DT>
  <DD>$Id: FastaParser.java 1246 2009-10-09 23:29:26Z soergel $</DD>
<DT><B>Author:</B></DT>
  <DD>David Tulga</DD>
</DL>
<HR>

<P>
<!-- =========== FIELD SUMMARY =========== -->

<A NAME="field_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Field Summary</B></FONT></TH>
</TR>
</TABLE>
&nbsp;<A NAME="fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Fields inherited from class edu.berkeley.compbio.sequtils.<A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#buf">buf</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#bufPosition">bufPosition</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<!-- ======== CONSTRUCTOR SUMMARY ======== -->

<A NAME="constructor_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Constructor Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, boolean)">FastaParser</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>&nbsp;fs,
            boolean&nbsp;skipGaps)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Initializes a parser from a fileset</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, int, boolean)">FastaParser</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>&nbsp;fs,
            int&nbsp;bufsize,
            boolean&nbsp;skipGaps)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Initializes a parser from a fileset and with a specific buffer size</TD>
</TR>
</TABLE>
&nbsp;
<!-- ========== METHOD SUMMARY =========== -->

<A NAME="method_summary"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
<B>Method Summary</B></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#close()">close</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Closes the reader, and all sub-readers</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getName()">getName</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the species name of this reader</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">getReader</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;section)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a SequenceReader at the given section</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, int)">getReader</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;section,
          int&nbsp;sectionPosition)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a SequenceReader at the given section</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getRootMetadata()">getRootMetadata</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getSequenceFragmentMetadata(java.lang.String)">getSequenceFragmentMetadata</A></B>(<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A>&nbsp;id)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTaxid()">getTaxid</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;long</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTotalSequence()">getTotalSequence</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the total amount of sequence present in this reader</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next section</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()">nextShuffled</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Iterates through the list of sections in random order, providing each section exactly once</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()">randomSectionFragment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionUniform()">randomSectionUniform</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()">read</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read(byte[], int)">read</A></B>(byte[]&nbsp;buffer,
     int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reads nucleotides into a byte array</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#reset()">reset</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;reset the stream to the beginning.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(long)">seek</A></B>(long&nbsp;newRawPosition)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;location)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)">seek</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;location,
     long&nbsp;offset)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">setFragmentLength</A></B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;frag)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A></CODE></FONT></TD>
<TD><CODE><B><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#toString()">toString</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class edu.berkeley.compbio.sequtils.<A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initMainBuffer(int)">initMainBuffer</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initTranslationBuffer(int)">initTranslationBuffer</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()">readTranslated</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationInvalid()">setTranslationInvalid</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_java.lang.Object"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from class java.lang.<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</A>, <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long, int)" title="class or interface in java.lang">wait</A></CODE></TD>
</TR>
</TABLE>
&nbsp;<A NAME="methods_inherited_from_class_edu.berkeley.compbio.sequtils.SequenceReader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#EEEEFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left"><B>Methods inherited from interface edu.berkeley.compbio.sequtils.<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></B></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated</A>, <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet</A></CODE></TD>
</TR>
</TABLE>
&nbsp;
<P>

<!-- ========= CONSTRUCTOR DETAIL ======== -->

<A NAME="constructor_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Constructor Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, boolean)"><!-- --></A><H3>
FastaParser</H3>
<PRE>
public <B>FastaParser</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>&nbsp;fs,
                   boolean&nbsp;skipGaps)</PRE>
<DL>
<DD>Initializes a parser from a fileset
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>fs</CODE> - The FastaFileSet</DL>
</DL>
<HR>

<A NAME="FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, int, boolean)"><!-- --></A><H3>
FastaParser</H3>
<PRE>
public <B>FastaParser</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</A>&nbsp;fs,
                   int&nbsp;bufsize,
                   boolean&nbsp;skipGaps)</PRE>
<DL>
<DD>Initializes a parser from a fileset and with a specific buffer size
<P>
<DL>
<DT><B>Parameters:</B><DD><CODE>fs</CODE> - The FastaFileSet<DD><CODE>bufsize</CODE> - The buffer size</DL>
</DL>

<!-- ============ METHOD DETAIL ========== -->

<A NAME="method_detail"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="1"><FONT SIZE="+2">
<B>Method Detail</B></FONT></TH>
</TR>
</TABLE>

<A NAME="getName()"><!-- --></A><H3>
getName</H3>
<PRE>
public <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A> <B>getName</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">SequenceReader</A></CODE></B></DD>
<DD>Returns the species name of this reader
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">getName</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>A String representing the name of the sequence</DL>
</DD>
</DL>
<HR>

<A NAME="close()"><!-- --></A><H3>
close</H3>
<PRE>
public void <B>close</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#close()">SequenceReader</A></CODE></B></DD>
<DD>Closes the reader, and all sub-readers
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#close()">close</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="toString()"><!-- --></A><H3>
toString</H3>
<PRE>
public <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A> <B>toString</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Overrides:</B><DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</A></CODE> in class <CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="randomSectionFragment()"><!-- --></A><H3>
randomSectionFragment</H3>
<PRE>
<FONT SIZE="-1">@NotNull
</FONT>public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>randomSectionFragment</B>()
                                               throws <A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A>,
                                                      <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">SectionList</A></CODE></B></DD>
<DD>Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.  Note this should not first select a random Section, since sections of
 different lengths may thereby produce biased results.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">randomSectionFragment</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></CODE>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="randomSectionLengthWeighted()"><!-- --></A><H3>
randomSectionLengthWeighted</H3>
<PRE>
public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>randomSectionLengthWeighted</B>()
                                                     throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD>Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The section
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="randomSectionUniform()"><!-- --></A><H3>
randomSectionUniform</H3>
<PRE>
public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>randomSectionUniform</B>()
                                              throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()">SectionList</A></CODE></B></DD>
<DD>Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()">randomSectionUniform</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="nextShuffled()"><!-- --></A><H3>
nextShuffled</H3>
<PRE>
<FONT SIZE="-1">@NotNull
</FONT>public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>nextShuffled</B>()
                                      throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A>,
                                             <A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">SectionList</A></CODE></B></DD>
<DD>Iterates through the list of sections in random order, providing each section exactly once
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">nextShuffled</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getTotalSequence()"><!-- --></A><H3>
getTotalSequence</H3>
<PRE>
public long <B>getTotalSequence</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">SequenceReader</A></CODE></B></DD>
<DD>Returns the total amount of sequence present in this reader
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">getTotalSequence</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The total amount of sequence</DL>
</DD>
</DL>
<HR>

<A NAME="read()"><!-- --></A><H3>
read</H3>
<PRE>
public byte <B>read</B>()
          throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A>,
                 <A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">SequenceReader</A></CODE></B></DD>
<DD>Returns the next nucleotide
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The next nucleotide, or EOF if the section or file has ended
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="read(byte[], int)"><!-- --></A><H3>
read</H3>
<PRE>
public int <B>read</B>(byte[]&nbsp;buffer,
                int&nbsp;length)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[], int)">SequenceReader</A></CODE></B></DD>
<DD>Reads nucleotides into a byte array
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[], int)">read</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>buffer</CODE> - a byte[] to read into.<DD><CODE>length</CODE> - how many nucleotides to read (must be less than or equal to the size of the given buffer)
<DT><B>Returns:</B><DD>The number of nucleotides read</DL>
</DD>
</DL>
<HR>

<A NAME="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><!-- --></A><H3>
seek</H3>
<PRE>
public void <B>seek</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;location)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">SequenceReader</A></CODE></B></DD>
<DD>Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="reset()"><!-- --></A><H3>
reset</H3>
<PRE>
public void <B>reset</B>()</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()">SectionList</A></CODE></B></DD>
<DD>reset the stream to the beginning.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#reset()">reset</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE><DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()">reset</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="next()"><!-- --></A><H3>
next</H3>
<PRE>
<FONT SIZE="-1">@NotNull
</FONT>public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>next</B>()
                              throws <A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A>,
                                     <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD>Returns the next section
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()">next</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>

<DT><B>Returns:</B><DD>The metadata of the next section
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="getRootMetadata()"><!-- --></A><H3>
getRootMetadata</H3>
<PRE>
public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>getRootMetadata</B>()</PRE>
<DL>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#getRootMetadata()">getRootMetadata</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><!-- --></A><H3>
setFragmentLength</H3>
<PRE>
public void <B>setFragmentLength</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;frag)
                       throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD>Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></CODE></DL>
</DD>
</DL>
<HR>

<A NAME="seek(long)"><!-- --></A><H3>
seek</H3>
<PRE>
protected void <B>seek</B>(long&nbsp;newRawPosition)</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)"><!-- --></A><H3>
getReader</H3>
<PRE>
public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A> <B>getReader</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;section)
                         throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD>Returns a SequenceReader at the given section
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>section</CODE> - The section to seek to
<DT><B>Returns:</B><DD>This, set to seek to the given section
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE> - If a file I/O problem is encountered</DL>
</DD>
</DL>
<HR>

<A NAME="getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, int)"><!-- --></A><H3>
getReader</H3>
<PRE>
public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A> <B>getReader</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;section,
                                int&nbsp;sectionPosition)
                         throws <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD>Returns a SequenceReader at the given section
<P>
<DD><DL>
</DL>
</DD>
<DD><DL>
<DT><B>Parameters:</B><DD><CODE>section</CODE> - The section to seek to<DD><CODE>sectionPosition</CODE> - The position within the section to seek to
<DT><B>Returns:</B><DD>This, set to seek to the given section
<DT><B>Throws:</B>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE> - If a file I/O problem is encountered</DL>
</DD>
</DL>
<HR>

<A NAME="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)"><!-- --></A><H3>
seek</H3>
<PRE>
public void <B>seek</B>(<A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A>&nbsp;location,
                 long&nbsp;offset)</PRE>
<DL>
<DD><B>Description copied from interface: <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)">SequenceReader</A></CODE></B></DD>
<DD>Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.
<P>
<DD><DL>
<DT><B>Specified by:</B><DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)">seek</A></CODE> in interface <CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getTaxid()"><!-- --></A><H3>
getTaxid</H3>
<PRE>
public <A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</A> <B>getTaxid</B>()</PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>
</DL>
</DD>
</DL>
<HR>

<A NAME="getSequenceFragmentMetadata(java.lang.String)"><!-- --></A><H3>
getSequenceFragmentMetadata</H3>
<PRE>
<FONT SIZE="-1">@NotNull
</FONT>public <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A> <B>getSequenceFragmentMetadata</B>(<A HREF="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</A>&nbsp;id)
                                                     throws <A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</A>,
                                                            <A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></PRE>
<DL>
<DD><DL>
</DL>
</DD>
<DD><DL>

<DT><B>Throws:</B>
<DD><CODE><A HREF="../../../../../../edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</A></CODE>
<DD><CODE><A HREF="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</A></CODE></DL>
</DD>
</DL>
<!-- ========= END OF CLASS DATA ========= -->
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Class</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="class-use/FastaParser.html"><FONT CLASS="NavBarFont1"><B>Use</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>PREV CLASS</B></A>&nbsp;
&nbsp;<A HREF="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>NEXT CLASS</B></A></FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../../index.html?edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="FastaParser.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
<TR>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
  SUMMARY:&nbsp;NESTED&nbsp;|&nbsp;<A HREF="#fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">FIELD</A>&nbsp;|&nbsp;<A HREF="#constructor_summary">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_summary">METHOD</A></FONT></TD>
<TD VALIGN="top" CLASS="NavBarCell3"><FONT SIZE="-2">
DETAIL:&nbsp;FIELD&nbsp;|&nbsp;<A HREF="#constructor_detail">CONSTR</A>&nbsp;|&nbsp;<A HREF="#method_detail">METHOD</A></FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.
</BODY>
</HTML>
