<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_60) on Tue Jul 08 13:00:17 EDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>FastaParser (sequtils 0.92-SNAPSHOT API)</title>
<meta name="date" content="2014-07-08">
<link rel="stylesheet" type="text/css" href="../../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="FastaParser (sequtils 0.92-SNAPSHOT API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" target="_top">Frames</a></li>
<li><a href="FastaParser.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<!-- ======== START OF CLASS DATA ======== -->
<div class="header">
<div class="subTitle">edu.berkeley.compbio.sequtils.sequencereader.fasta</div>
<h2 title="Class FastaParser" class="title">Class FastaParser</h2>
</div>
<div class="contentContainer">
<ul class="inheritance">
<li><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">java.lang.Object</a></li>
<li>
<ul class="inheritance">
<li><a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">edu.berkeley.compbio.sequtils.TranslatingSequenceReader</a></li>
<li>
<ul class="inheritance">
<li>edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaParser</li>
</ul>
</li>
</ul>
</li>
</ul>
<div class="description">
<ul class="blockList">
<li class="blockList">
<dl>
<dt>All Implemented Interfaces:</dt>
<dd><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></dd>
</dl>
<hr>
<br>
<pre>public class <span class="strong">FastaParser</span>
extends <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a>
implements <a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></pre>
<div class="block">Fasta File Parser</div>
<dl><dt><span class="strong">Version:</span></dt>
  <dd>$Id: FastaParser.java 1246 2009-10-09 23:29:26Z soergel $</dd>
<dt><span class="strong">Author:</span></dt>
  <dd>David Tulga</dd></dl>
</li>
</ul>
</div>
<div class="summary">
<ul class="blockList">
<li class="blockList">
<!-- =========== FIELD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="field_summary">
<!--   -->
</a>
<h3>Field Summary</h3>
<ul class="blockList">
<li class="blockList"><a name="fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">
<!--   -->
</a>
<h3>Fields inherited from class&nbsp;edu.berkeley.compbio.sequtils.<a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></h3>
<code><a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#buf">buf</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#bufPosition">bufPosition</a></code></li>
</ul>
</li>
</ul>
<!-- ======== CONSTRUCTOR SUMMARY ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_summary">
<!--   -->
</a>
<h3>Constructor Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Constructor Summary table, listing constructors, and an explanation">
<caption><span>Constructors</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colOne" scope="col">Constructor and Description</th>
</tr>
<tr class="altColor">
<td class="colOne"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet,%20boolean)">FastaParser</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a>&nbsp;fs,
           boolean&nbsp;skipGaps)</code>
<div class="block">Initializes a parser from a fileset</div>
</td>
</tr>
<tr class="rowColor">
<td class="colOne"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet,%20int,%20boolean)">FastaParser</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a>&nbsp;fs,
           int&nbsp;bufsize,
           boolean&nbsp;skipGaps)</code>
<div class="block">Initializes a parser from a fileset and with a specific buffer size</div>
</td>
</tr>
</table>
</li>
</ul>
<!-- ========== METHOD SUMMARY =========== -->
<ul class="blockList">
<li class="blockList"><a name="method_summary">
<!--   -->
</a>
<h3>Method Summary</h3>
<table class="overviewSummary" border="0" cellpadding="3" cellspacing="0" summary="Method Summary table, listing methods, and an explanation">
<caption><span>Methods</span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#close()">close</a></strong>()</code>
<div class="block">Closes the reader, and all sub-readers</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getName()">getName</a></strong>()</code>
<div class="block">Returns the species name of this reader</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">getReader</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section)</code>
<div class="block">Returns a SequenceReader at the given section</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20int)">getReader</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section,
         int&nbsp;sectionPosition)</code>
<div class="block">Returns a SequenceReader at the given section</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getRootMetadata()">getRootMetadata</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getSequenceFragmentMetadata(java.lang.String)">getSequenceFragmentMetadata</a></strong>(<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTaxid()">getTaxid</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>long</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#getTotalSequence()">getTotalSequence</a></strong>()</code>
<div class="block">Returns the total amount of sequence present in this reader</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()">next</a></strong>()</code>
<div class="block">Returns the next section</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()">nextShuffled</a></strong>()</code>
<div class="block">Iterates through the list of sections in random order, providing each section exactly once</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()">randomSectionFragment</a></strong>()</code>
<div class="block">Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</a></strong>()</code>
<div class="block">Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionUniform()">randomSectionUniform</a></strong>()</code>
<div class="block">Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()">read</a></strong>()</code>
<div class="block">Returns the next nucleotide</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read(byte[],%20int)">read</a></strong>(byte[]&nbsp;buffer,
    int&nbsp;length)</code>
<div class="block">Reads nucleotides into a byte array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#reset()">reset</a></strong>()</code>
<div class="block">reset the stream to the beginning.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(long)">seek</a></strong>(long&nbsp;newRawPosition)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;location)</code>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;location,
    long&nbsp;offset)</code>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">setFragmentLength</a></strong>(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;frag)</code>
<div class="block">Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a></code></td>
<td class="colLast"><code><strong><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#toString()">toString</a></strong>()</code>&nbsp;</td>
</tr>
</table>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;edu.berkeley.compbio.sequtils.<a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html" title="class in edu.berkeley.compbio.sequtils">TranslatingSequenceReader</a></h3>
<code><a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initMainBuffer(int)">initMainBuffer</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#initTranslationBuffer(int)">initTranslationBuffer</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()">readTranslated</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#setTranslationInvalid()">setTranslationInvalid</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_java.lang.Object">
<!--   -->
</a>
<h3>Methods inherited from class&nbsp;java.lang.<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></h3>
<code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#clone()" title="class or interface in java.lang">clone</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#equals(java.lang.Object)" title="class or interface in java.lang">equals</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#finalize()" title="class or interface in java.lang">finalize</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#getClass()" title="class or interface in java.lang">getClass</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#hashCode()" title="class or interface in java.lang">hashCode</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notify()" title="class or interface in java.lang">notify</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#notifyAll()" title="class or interface in java.lang">notifyAll</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait()" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long)" title="class or interface in java.lang">wait</a>, <a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#wait(long,%20int)" title="class or interface in java.lang">wait</a></code></li>
</ul>
<ul class="blockList">
<li class="blockList"><a name="methods_inherited_from_class_edu.berkeley.compbio.sequtils.SequenceReader">
<!--   -->
</a>
<h3>Methods inherited from interface&nbsp;edu.berkeley.compbio.sequtils.<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></h3>
<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated</a>, <a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#setTranslationAlphabet(byte[])">setTranslationAlphabet</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
<div class="details">
<ul class="blockList">
<li class="blockList">
<!-- ========= CONSTRUCTOR DETAIL ======== -->
<ul class="blockList">
<li class="blockList"><a name="constructor_detail">
<!--   -->
</a>
<h3>Constructor Detail</h3>
<a name="FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, boolean)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>FastaParser</h4>
<pre>public&nbsp;FastaParser(<a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a>&nbsp;fs,
           boolean&nbsp;skipGaps)</pre>
<div class="block">Initializes a parser from a fileset</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fs</code> - The FastaFileSet</dd></dl>
</li>
</ul>
<a name="FastaParser(edu.berkeley.compbio.sequtils.sequencereader.fasta.FastaFileSet, int, boolean)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>FastaParser</h4>
<pre>public&nbsp;FastaParser(<a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta">FastaFileSet</a>&nbsp;fs,
           int&nbsp;bufsize,
           boolean&nbsp;skipGaps)</pre>
<div class="block">Initializes a parser from a fileset and with a specific buffer size</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>fs</code> - The FastaFileSet</dd><dd><code>bufsize</code> - The buffer size</dd></dl>
</li>
</ul>
</li>
</ul>
<!-- ============ METHOD DETAIL ========== -->
<ul class="blockList">
<li class="blockList"><a name="method_detail">
<!--   -->
</a>
<h3>Method Detail</h3>
<a name="getName()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getName</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;getName()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">SequenceReader</a></code></strong></div>
<div class="block">Returns the species name of this reader</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getName()">getName</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>A String representing the name of the sequence</dd></dl>
</li>
</ul>
<a name="close()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>close</h4>
<pre>public&nbsp;void&nbsp;close()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#close()">SequenceReader</a></code></strong></div>
<div class="block">Closes the reader, and all sub-readers</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#close()">close</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
</dl>
</li>
</ul>
<a name="toString()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>toString</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;toString()</pre>
<dl>
<dt><strong>Overrides:</strong></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true#toString()" title="class or interface in java.lang">toString</a></code>&nbsp;in class&nbsp;<code><a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Object.html?is-external=true" title="class or interface in java.lang">Object</a></code></dd>
</dl>
</li>
</ul>
<a name="randomSectionFragment()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>randomSectionFragment</h4>
<pre>@NotNull
public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;randomSectionFragment()
                                               throws <a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a>,
                                                      <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">SectionList</a></code></strong></div>
<div class="block">Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.  Note this should not first select a random Section, since sections of
 different lengths may thereby produce biased results.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">randomSectionFragment</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="randomSectionLengthWeighted()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>randomSectionLengthWeighted</h4>
<pre>public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;randomSectionLengthWeighted()
                                                     throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Return a section, randomly chosen nucleotidewise (i.e., weighted by section length).</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The section</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="randomSectionUniform()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>randomSectionUniform</h4>
<pre>public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;randomSectionUniform()
                                              throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()">SectionList</a></code></strong></div>
<div class="block">Get a random section from the list, selected uniformly from the complete sections in the list, ignoring their
 lengths.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionUniform()">randomSectionUniform</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
<a name="nextShuffled()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>nextShuffled</h4>
<pre>@NotNull
public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;nextShuffled()
                                      throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                                             <a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">SectionList</a></code></strong></div>
<div class="block">Iterates through the list of sections in random order, providing each section exactly once</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">nextShuffled</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd></dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="getTotalSequence()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTotalSequence</h4>
<pre>public&nbsp;long&nbsp;getTotalSequence()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">SequenceReader</a></code></strong></div>
<div class="block">Returns the total amount of sequence present in this reader</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#getTotalSequence()">getTotalSequence</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The total amount of sequence</dd></dl>
</li>
</ul>
<a name="read()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>public&nbsp;byte&nbsp;read()
          throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a>,
                 <a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">SequenceReader</a></code></strong></div>
<div class="block">Returns the next nucleotide</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The next nucleotide, or EOF if the section or file has ended</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="read(byte[], int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>read</h4>
<pre>public&nbsp;int&nbsp;read(byte[]&nbsp;buffer,
       int&nbsp;length)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[],%20int)">SequenceReader</a></code></strong></div>
<div class="block">Reads nucleotides into a byte array</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[],%20int)">read</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
<dt><span class="strong">Parameters:</span></dt><dd><code>buffer</code> - a byte[] to read into.</dd><dd><code>length</code> - how many nucleotides to read (must be less than or equal to the size of the given buffer)</dd>
<dt><span class="strong">Returns:</span></dt><dd>The number of nucleotides read</dd></dl>
</li>
</ul>
<a name="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seek</h4>
<pre>public&nbsp;void&nbsp;seek(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;location)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">SequenceReader</a></code></strong></div>
<div class="block">Set the location of the reader to that specified by the given SequenceFragmentMetadata, so that the next read()
 operation will return the first character of the specified SequenceFragment.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">seek</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
</dl>
</li>
</ul>
<a name="reset()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>reset</h4>
<pre>public&nbsp;void&nbsp;reset()</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()">SectionList</a></code></strong></div>
<div class="block">reset the stream to the beginning.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#reset()">reset</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#reset()">reset</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
</dl>
</li>
</ul>
<a name="next()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>next</h4>
<pre>@NotNull
public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;next()
                              throws <a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a>,
                                     <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Returns the next section</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()">next</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
<dt><span class="strong">Returns:</span></dt><dd>The metadata of the next section</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="getRootMetadata()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getRootMetadata</h4>
<pre>public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;getRootMetadata()</pre>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#getRootMetadata()">getRootMetadata</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html" title="interface in edu.berkeley.compbio.sequtils.sequencereader">SectionList</a></code></dd>
</dl>
</li>
</ul>
<a name="setFragmentLength(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>setFragmentLength</h4>
<pre>public&nbsp;void&nbsp;setFragmentLength(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;frag)
                       throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Returns a SequenceFragmentMetadata encompassing all sequence available to this parser, inheriting from the parent
 metadata of the same range for lame reasons</div>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></code></dd></dl>
</li>
</ul>
<a name="seek(long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seek</h4>
<pre>protected&nbsp;void&nbsp;seek(long&nbsp;newRawPosition)</pre>
</li>
</ul>
<a name="getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReader</h4>
<pre>public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a>&nbsp;getReader(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section)
                         throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Returns a SequenceReader at the given section</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>section</code> - The section to seek to</dd>
<dt><span class="strong">Returns:</span></dt><dd>This, set to seek to the given section</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - If a file I/O problem is encountered</dd></dl>
</li>
</ul>
<a name="getReader(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, int)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getReader</h4>
<pre>public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a>&nbsp;getReader(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;section,
                       int&nbsp;sectionPosition)
                         throws <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<div class="block">Returns a SequenceReader at the given section</div>
<dl><dt><span class="strong">Parameters:</span></dt><dd><code>section</code> - The section to seek to</dd><dd><code>sectionPosition</code> - The position within the section to seek to</dd>
<dt><span class="strong">Returns:</span></dt><dd>This, set to seek to the given section</dd>
<dt><span class="strong">Throws:</span></dt>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code> - If a file I/O problem is encountered</dd></dl>
</li>
</ul>
<a name="seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata, long)">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>seek</h4>
<pre>public&nbsp;void&nbsp;seek(<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;location,
        long&nbsp;offset)</pre>
<div class="block"><strong>Description copied from interface:&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">SequenceReader</a></code></strong></div>
<div class="block">Set the location of the reader to an offset beyond the location specified by the given SequenceFragmentMetadata, so
 that the next read() operation will return the character at that offset of the specified SequenceFragment.</div>
<dl>
<dt><strong>Specified by:</strong></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#seek(edu.berkeley.compbio.sequtils.SequenceFragmentMetadata,%20long)">seek</a></code>&nbsp;in interface&nbsp;<code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></dd>
</dl>
</li>
</ul>
<a name="getTaxid()">
<!--   -->
</a>
<ul class="blockList">
<li class="blockList">
<h4>getTaxid</h4>
<pre>public&nbsp;<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/Integer.html?is-external=true" title="class or interface in java.lang">Integer</a>&nbsp;getTaxid()</pre>
</li>
</ul>
<a name="getSequenceFragmentMetadata(java.lang.String)">
<!--   -->
</a>
<ul class="blockListLast">
<li class="blockList">
<h4>getSequenceFragmentMetadata</h4>
<pre>@NotNull
public&nbsp;<a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a>&nbsp;getSequenceFragmentMetadata(<a href="http://docs.oracle.com/javase/6/docs/api/java/lang/String.html?is-external=true" title="class or interface in java.lang">String</a>&nbsp;id)
                                                     throws <a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</a>,
                                                            <a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></pre>
<dl><dt><span class="strong">Throws:</span></dt>
<dd><code><a href="../../../../../../edu/berkeley/compbio/sequtils/SequenceException.html" title="class in edu.berkeley.compbio.sequtils">SequenceException</a></code></dd>
<dd><code><a href="http://docs.oracle.com/javase/6/docs/api/java/io/IOException.html?is-external=true" title="class or interface in java.io">IOException</a></code></dd></dl>
</li>
</ul>
</li>
</ul>
</li>
</ul>
</div>
</div>
<!-- ========= END OF CLASS DATA ========= -->
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../../overview-summary.html">Overview</a></li>
<li><a href="package-summary.html">Package</a></li>
<li class="navBarCell1Rev">Class</li>
<li><a href="class-use/FastaParser.html">Use</a></li>
<li><a href="package-tree.html">Tree</a></li>
<li><a href="../../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">Prev Class</span></a></li>
<li><a href="../../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParserSet.html" title="class in edu.berkeley.compbio.sequtils.sequencereader.fasta"><span class="strong">Next Class</span></a></li>
</ul>
<ul class="navList">
<li><a href="../../../../../../index.html?edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html" target="_top">Frames</a></li>
<li><a href="FastaParser.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<div>
<ul class="subNavList">
<li>Summary:&nbsp;</li>
<li>Nested&nbsp;|&nbsp;</li>
<li><a href="#fields_inherited_from_class_edu.berkeley.compbio.sequtils.TranslatingSequenceReader">Field</a>&nbsp;|&nbsp;</li>
<li><a href="#constructor_summary">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_summary">Method</a></li>
</ul>
<ul class="subNavList">
<li>Detail:&nbsp;</li>
<li>Field&nbsp;|&nbsp;</li>
<li><a href="#constructor_detail">Constr</a>&nbsp;|&nbsp;</li>
<li><a href="#method_detail">Method</a></li>
</ul>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.</small></p>
</body>
</html>
