<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!-- NewPage -->
<html lang="en">
<head>
<!-- Generated by javadoc (version 1.7.0_60) on Tue Jul 08 13:00:17 EDT 2014 -->
<meta http-equiv="Content-Type" content="text/html" charset="UTF-8">
<title>Uses of Class edu.berkeley.compbio.sequtils.NotEnoughSequenceException (sequtils 0.92-SNAPSHOT API)</title>
<meta name="date" content="2014-07-08">
<link rel="stylesheet" type="text/css" href="../../../../../stylesheet.css" title="Style">
</head>
<body>
<script type="text/javascript"><!--
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class edu.berkeley.compbio.sequtils.NotEnoughSequenceException (sequtils 0.92-SNAPSHOT API)";
    }
//-->
</script>
<noscript>
<div>JavaScript is disabled on your browser.</div>
</noscript>
<!-- ========= START OF TOP NAVBAR ======= -->
<div class="topNav"><a name="navbar_top">
<!--   -->
</a><a href="#skip-navbar_top" title="Skip navigation links"></a><a name="navbar_top_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?edu/berkeley/compbio/sequtils/class-use/NotEnoughSequenceException.html" target="_top">Frames</a></li>
<li><a href="NotEnoughSequenceException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_top">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_top");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_top">
<!--   -->
</a></div>
<!-- ========= END OF TOP NAVBAR ========= -->
<div class="header">
<h2 title="Uses of Class edu.berkeley.compbio.sequtils.NotEnoughSequenceException" class="title">Uses of Class<br>edu.berkeley.compbio.sequtils.NotEnoughSequenceException</h2>
</div>
<div class="classUseContainer">
<ul class="blockList">
<li class="blockList">
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing packages, and an explanation">
<caption><span>Packages that use <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Package</th>
<th class="colLast" scope="col">Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><a href="#edu.berkeley.compbio.sequtils">edu.berkeley.compbio.sequtils</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#edu.berkeley.compbio.sequtils.sequencefragmentiterator">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#edu.berkeley.compbio.sequtils.sequencereader">edu.berkeley.compbio.sequtils.sequencereader</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><a href="#edu.berkeley.compbio.sequtils.sequencereader.fasta">edu.berkeley.compbio.sequtils.sequencereader.fasta</a></td>
<td class="colLast">&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><a href="#edu.berkeley.compbio.sequtils.strings">edu.berkeley.compbio.sequtils.strings</a></td>
<td class="colLast">&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList">
<ul class="blockList">
<li class="blockList"><a name="edu.berkeley.compbio.sequtils">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a> in <a href="../../../../../edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</a> that throw <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>static byte[]</code></td>
<td class="colLast"><span class="strong">SequenceArrayUtils.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySliceUpToNNonGaps(byte[],%20int,%20int)">copySliceUpToNNonGaps</a></strong>(byte[]&nbsp;x,
                     int&nbsp;pos,
                     int&nbsp;nonGapsDesired)</code>
<div class="block">copy a slice starting at the given position of whatever width is needed to obtain the requested number of non-gap
 characters</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><span class="strong">SequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read</a></strong>()</code>
<div class="block">Returns the next nucleotide</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><span class="strong">FilteringSequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read()">read</a></strong>()</code>
<div class="block">Returns the next nucleotide</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><span class="strong">ByteArraySequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read()">read</a></strong>()</code>
<div class="block">Returns the next nucleotide</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">SequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[],%20int)">read</a></strong>(byte[]&nbsp;buffer,
    int&nbsp;length)</code>
<div class="block">Reads nucleotides into a byte array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">FilteringSequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read(byte[],%20int)">read</a></strong>(byte[]&nbsp;buffer,
    int&nbsp;length)</code>
<div class="block">Reads nucleotides into a byte array</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">ByteArraySequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read(byte[],%20int)">read</a></strong>(byte[]&nbsp;buffer,
    int&nbsp;length)</code>
<div class="block">Reads nucleotides into a byte array</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">SequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated</a></strong>()</code>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">FilteringSequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#readTranslated()">readTranslated</a></strong>()</code>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">TranslatingSequenceReader.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()">readTranslated</a></strong>()</code>
<div class="block">Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="edu.berkeley.compbio.sequtils.sequencefragmentiterator">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a> in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</a> that throw <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">SectionDecomposingSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow()">getNextWindow</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">SectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow()">getNextWindow</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">ShuffledSectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow()">getNextWindow</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">RandomSectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow()">getNextWindow</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected abstract <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">SectionDecomposingSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow(int)">getNextWindow</a></strong>(int&nbsp;theWidth)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">SectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow(int)">getNextWindow</a></strong>(int&nbsp;theWidth)</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">ShuffledSectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow(int)">getNextWindow</a></strong>(int&nbsp;theWidth)</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected <a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a></code></td>
<td class="colLast"><span class="strong">RandomSectionSFI.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow(int)">getNextWindow</a></strong>(int&nbsp;theWidth)</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="edu.berkeley.compbio.sequtils.sequencereader">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a> in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</a> that throw <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FilteringSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#next()">next</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">SectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()">next</a></strong>()</code>
<div class="block">Iterates through the list of sections in order.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FilteringSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#nextShuffled()">nextShuffled</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">SectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">nextShuffled</a></strong>()</code>
<div class="block">Iterates through the list of sections in random order, providing each section exactly once</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FilteringSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionFragment()">randomSectionFragment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">SectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">randomSectionFragment</a></strong>()</code>
<div class="block">Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">RandomSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionFragment()">randomSectionFragment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FilteringSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">SectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</a></strong>()</code>
<div class="block">Get a random section from the list, selected uniformly from the complete sections in the list, weighted by their
 lengths.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">RandomSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">RandomSectionList.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#readTranslated()">readTranslated</a></strong>()</code>&nbsp;</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="edu.berkeley.compbio.sequtils.sequencereader.fasta">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a> in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</a> that throw <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FastaParser.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()">next</a></strong>()</code>
<div class="block">Returns the next section</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FastaParser.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()">nextShuffled</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</a></code></td>
<td class="colLast"><span class="strong">FastaParser.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()">randomSectionFragment</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>byte</code></td>
<td class="colLast"><span class="strong">FastaParser.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()">read</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code>int</code></td>
<td class="colLast"><span class="strong">FastaFileSet.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#readAt(byte[],%20long,%20int)">readAt</a></strong>(byte[]&nbsp;buf,
      long&nbsp;position,
      int&nbsp;length)</code>
<div class="block">Reads a section from the files</div>
</td>
</tr>
</tbody>
</table>
</li>
<li class="blockList"><a name="edu.berkeley.compbio.sequtils.strings">
<!--   -->
</a>
<h3>Uses of <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a> in <a href="../../../../../edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a></h3>
<table border="0" cellpadding="3" cellspacing="0" summary="Use table, listing methods, and an explanation">
<caption><span>Methods in <a href="../../../../../edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</a> that throw <a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</a></span><span class="tabEnd">&nbsp;</span></caption>
<tr>
<th class="colFirst" scope="col">Modifier and Type</th>
<th class="colLast" scope="col">Method and Description</th>
</tr>
<tbody>
<tr class="altColor">
<td class="colFirst"><code>void</code></td>
<td class="colLast"><span class="strong">SequenceFragment.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#checkAvailable()">checkAvailable</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></td>
<td class="colLast"><span class="strong">AggregateSequenceFragment.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getResetReader()">getResetReader</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</a></code></td>
<td class="colLast"><span class="strong">SequenceFragment.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getResetReader()">getResetReader</a></strong>()</code>
<div class="block">Returns the number of samples on which the base spectrum is based.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code>protected void</code></td>
<td class="colLast"><span class="strong">SequenceFragment.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanSpectrum()">rescanSpectrum</a></strong>()</code>&nbsp;</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></code></td>
<td class="colLast"><span class="strong">SequenceSpectrumScanner.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence</a></strong>(<a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a>&nbsp;fragment)</code>
<div class="block">Scans a sequence to count pattern frequencies.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></code></td>
<td class="colLast"><span class="strong">KcountScanner.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence</a></strong>(<a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a>&nbsp;fragment)</code>
<div class="block">Scans a sequence to count pattern frequencies.</div>
</td>
</tr>
<tr class="altColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</a></code></td>
<td class="colLast"><span class="strong">SequenceSpectrumScanner.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20byte[])">scanSequence</a></strong>(<a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a>&nbsp;fragment,
            byte[]&nbsp;prefix)</code>
<div class="block">Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</div>
</td>
</tr>
<tr class="rowColor">
<td class="colFirst"><code><a href="../../../../../edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</a></code></td>
<td class="colLast"><span class="strong">KcountScanner.</span><code><strong><a href="../../../../../edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment,%20byte[])">scanSequence</a></strong>(<a href="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</a>&nbsp;fragment,
            byte[]&nbsp;prefix)</code>
<div class="block">Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</div>
</td>
</tr>
</tbody>
</table>
</li>
</ul>
</li>
</ul>
</div>
<!-- ======= START OF BOTTOM NAVBAR ====== -->
<div class="bottomNav"><a name="navbar_bottom">
<!--   -->
</a><a href="#skip-navbar_bottom" title="Skip navigation links"></a><a name="navbar_bottom_firstrow">
<!--   -->
</a>
<ul class="navList" title="Navigation">
<li><a href="../../../../../overview-summary.html">Overview</a></li>
<li><a href="../package-summary.html">Package</a></li>
<li><a href="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">Class</a></li>
<li class="navBarCell1Rev">Use</li>
<li><a href="../package-tree.html">Tree</a></li>
<li><a href="../../../../../deprecated-list.html">Deprecated</a></li>
<li><a href="../../../../../index-all.html">Index</a></li>
<li><a href="../../../../../help-doc.html">Help</a></li>
</ul>
</div>
<div class="subNav">
<ul class="navList">
<li>Prev</li>
<li>Next</li>
</ul>
<ul class="navList">
<li><a href="../../../../../index.html?edu/berkeley/compbio/sequtils/class-use/NotEnoughSequenceException.html" target="_top">Frames</a></li>
<li><a href="NotEnoughSequenceException.html" target="_top">No Frames</a></li>
</ul>
<ul class="navList" id="allclasses_navbar_bottom">
<li><a href="../../../../../allclasses-noframe.html">All Classes</a></li>
</ul>
<div>
<script type="text/javascript"><!--
  allClassesLink = document.getElementById("allclasses_navbar_bottom");
  if(window==top) {
    allClassesLink.style.display = "block";
  }
  else {
    allClassesLink.style.display = "none";
  }
  //-->
</script>
</div>
<a name="skip-navbar_bottom">
<!--   -->
</a></div>
<!-- ======== END OF BOTTOM NAVBAR ======= -->
<p class="legalCopy"><small>Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.</small></p>
</body>
</html>
