<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN" "http://www.w3.org/TR/html4/loose.dtd">
<!--NewPage-->
<HTML>
<HEAD>
<!-- Generated by javadoc (build 1.6.0_27) on Sun Jul 06 18:05:53 EDT 2014 -->
<META http-equiv="Content-Type" content="text/html; charset=UTF-8">
<TITLE>
Uses of Class edu.berkeley.compbio.sequtils.NotEnoughSequenceException (sequtils 0.92-SNAPSHOT API)
</TITLE>

<META NAME="date" CONTENT="2014-07-06">

<LINK REL ="stylesheet" TYPE="text/css" HREF="../../../../../stylesheet.css" TITLE="Style">

<SCRIPT type="text/javascript">
function windowTitle()
{
    if (location.href.indexOf('is-external=true') == -1) {
        parent.document.title="Uses of Class edu.berkeley.compbio.sequtils.NotEnoughSequenceException (sequtils 0.92-SNAPSHOT API)";
    }
}
</SCRIPT>
<NOSCRIPT>
</NOSCRIPT>

</HEAD>

<BODY BGCOLOR="white" onload="windowTitle();">
<HR>


<!-- ========= START OF TOP NAVBAR ======= -->
<A NAME="navbar_top"><!-- --></A>
<A HREF="#skip-navbar_top" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_top_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?edu/berkeley/compbio/sequtils//class-useNotEnoughSequenceException.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NotEnoughSequenceException.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_top"></A>
<!-- ========= END OF TOP NAVBAR ========= -->

<HR>
<CENTER>
<H2>
<B>Uses of Class<br>edu.berkeley.compbio.sequtils.NotEnoughSequenceException</B></H2>
</CENTER>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Packages that use <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#edu.berkeley.compbio.sequtils"><B>edu.berkeley.compbio.sequtils</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#edu.berkeley.compbio.sequtils.sequencefragmentiterator"><B>edu.berkeley.compbio.sequtils.sequencefragmentiterator</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#edu.berkeley.compbio.sequtils.sequencereader"><B>edu.berkeley.compbio.sequtils.sequencereader</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#edu.berkeley.compbio.sequtils.sequencereader.fasta"><B>edu.berkeley.compbio.sequtils.sequencereader.fasta</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD><A HREF="#edu.berkeley.compbio.sequtils.strings"><B>edu.berkeley.compbio.sequtils.strings</B></A></TD>
<TD>&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="edu.berkeley.compbio.sequtils"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A> in <A HREF="../../../../../edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../edu/berkeley/compbio/sequtils/package-summary.html">edu.berkeley.compbio.sequtils</A> that throw <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>static&nbsp;byte[]</CODE></FONT></TD>
<TD><CODE><B>SequenceArrayUtils.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceArrayUtils.html#copySliceUpToNNonGaps(byte[], int, int)">copySliceUpToNNonGaps</A></B>(byte[]&nbsp;x,
                      int&nbsp;pos,
                      int&nbsp;nonGapsDesired)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;copy a slice starting at the given position of whatever width is needed to obtain the requested number of non-gap
 characters</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B>SequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read()">read</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B>FilteringSequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read()">read</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B>ByteArraySequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read()">read</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#read(byte[], int)">read</A></B>(byte[]&nbsp;buffer,
     int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reads nucleotides into a byte array</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>FilteringSequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#read(byte[], int)">read</A></B>(byte[]&nbsp;buffer,
     int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reads nucleotides into a byte array</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>ByteArraySequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/ByteArraySequenceReader.html#read(byte[], int)">read</A></B>(byte[]&nbsp;buffer,
     int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reads nucleotides into a byte array</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>SequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html#readTranslated()">readTranslated</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>FilteringSequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/FilteringSequenceReader.html#readTranslated()">readTranslated</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>TranslatingSequenceReader.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/TranslatingSequenceReader.html#readTranslated()">readTranslated</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next nucleotide, after mapping the byte through a translation table associated with the reader.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="edu.berkeley.compbio.sequtils.sequencefragmentiterator"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A> in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/package-summary.html">edu.berkeley.compbio.sequtils.sequencefragmentiterator</A> that throw <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>SectionDecomposingSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow()">getNextWindow</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>SectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow()">getNextWindow</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>ShuffledSectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow()">getNextWindow</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>RandomSectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow()">getNextWindow</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected abstract &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>SectionDecomposingSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionDecomposingSFI.html#getNextWindow(int)">getNextWindow</A></B>(int&nbsp;theWidth)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>SectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/SectionSFI.html#getNextWindow(int)">getNextWindow</A></B>(int&nbsp;theWidth)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>ShuffledSectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/ShuffledSectionSFI.html#getNextWindow(int)">getNextWindow</A></B>(int&nbsp;theWidth)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A></CODE></FONT></TD>
<TD><CODE><B>RandomSectionSFI.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencefragmentiterator/RandomSectionSFI.html#getNextWindow(int)">getNextWindow</A></B>(int&nbsp;theWidth)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="edu.berkeley.compbio.sequtils.sequencereader"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A> in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader</A> that throw <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FilteringSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>SectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Iterates through the list of sections in order.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FilteringSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#nextShuffled()">nextShuffled</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>SectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#nextShuffled()">nextShuffled</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Iterates through the list of sections in random order, providing each section exactly once</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FilteringSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionFragment()">randomSectionFragment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>SectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionFragment()">randomSectionFragment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get a random section fragment (really just the start position, with no length specified), selected uniformly
 (nucleotidewise) from all available sequence.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>RandomSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionFragment()">randomSectionFragment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FilteringSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/FilteringSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>SectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/SectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Get a random section from the list, selected uniformly from the complete sections in the list, weighted by their
 lengths.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>RandomSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#randomSectionLengthWeighted()">randomSectionLengthWeighted</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>RandomSectionList.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/RandomSectionList.html#readTranslated()">readTranslated</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="edu.berkeley.compbio.sequtils.sequencereader.fasta"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A> in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/package-summary.html">edu.berkeley.compbio.sequtils.sequencereader.fasta</A> that throw <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FastaParser.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#next()">next</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the next section</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FastaParser.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#nextShuffled()">nextShuffled</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceFragmentMetadata.html" title="class in edu.berkeley.compbio.sequtils">SequenceFragmentMetadata</A></CODE></FONT></TD>
<TD><CODE><B>FastaParser.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#randomSectionFragment()">randomSectionFragment</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;byte</CODE></FONT></TD>
<TD><CODE><B>FastaParser.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaParser.html#read()">read</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;int</CODE></FONT></TD>
<TD><CODE><B>FastaFileSet.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/sequencereader/fasta/FastaFileSet.html#readAt(byte[], long, int)">readAt</A></B>(byte[]&nbsp;buf,
       long&nbsp;position,
       int&nbsp;length)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Reads a section from the files</TD>
</TR>
</TABLE>
&nbsp;
<P>
<A NAME="edu.berkeley.compbio.sequtils.strings"><!-- --></A>
<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableHeadingColor">
<TH ALIGN="left" COLSPAN="2"><FONT SIZE="+2">
Uses of <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A> in <A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A></FONT></TH>
</TR>
</TABLE>
&nbsp;
<P>

<TABLE BORDER="1" WIDTH="100%" CELLPADDING="3" CELLSPACING="0" SUMMARY="">
<TR BGCOLOR="#CCCCFF" CLASS="TableSubHeadingColor">
<TH ALIGN="left" COLSPAN="2">Methods in <A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/package-summary.html">edu.berkeley.compbio.sequtils.strings</A> that throw <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils">NotEnoughSequenceException</A></FONT></TH>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SequenceFragment.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#checkAvailable()">checkAvailable</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></FONT></TD>
<TD><CODE><B>AggregateSequenceFragment.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/AggregateSequenceFragment.html#getResetReader()">getResetReader</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/SequenceReader.html" title="interface in edu.berkeley.compbio.sequtils">SequenceReader</A></CODE></FONT></TD>
<TD><CODE><B>SequenceFragment.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#getResetReader()">getResetReader</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Returns the number of samples on which the base spectrum is based.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>protected &nbsp;void</CODE></FONT></TD>
<TD><CODE><B>SequenceFragment.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html#rescanSpectrum()">rescanSpectrum</A></B>()</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A></CODE></FONT></TD>
<TD><CODE><B>SequenceSpectrumScanner.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence</A></B>(<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>&nbsp;fragment)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans a sequence to count pattern frequencies.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A></CODE></FONT></TD>
<TD><CODE><B>KcountScanner.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment)">scanSequence</A></B>(<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>&nbsp;fragment)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans a sequence to count pattern frequencies.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrum.html" title="interface in edu.berkeley.compbio.sequtils.strings">SequenceSpectrum</A></CODE></FONT></TD>
<TD><CODE><B>SequenceSpectrumScanner.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceSpectrumScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment, byte[])">scanSequence</A></B>(<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>&nbsp;fragment,
             byte[]&nbsp;prefix)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</TD>
</TR>
<TR BGCOLOR="white" CLASS="TableRowColor">
<TD ALIGN="right" VALIGN="top" WIDTH="1%"><FONT SIZE="-1">
<CODE>&nbsp;<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/Kcount.html" title="class in edu.berkeley.compbio.sequtils.strings">Kcount</A></CODE></FONT></TD>
<TD><CODE><B>KcountScanner.</B><B><A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/KcountScanner.html#scanSequence(edu.berkeley.compbio.sequtils.strings.SequenceFragment, byte[])">scanSequence</A></B>(<A HREF="../../../../../edu/berkeley/compbio/sequtils/strings/SequenceFragment.html" title="class in edu.berkeley.compbio.sequtils.strings">SequenceFragment</A>&nbsp;fragment,
             byte[]&nbsp;prefix)</CODE>

<BR>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Scans a sequence to count pattern frequencies, considering only words that follow the given prefix.</TD>
</TR>
</TABLE>
&nbsp;
<P>
<HR>


<!-- ======= START OF BOTTOM NAVBAR ====== -->
<A NAME="navbar_bottom"><!-- --></A>
<A HREF="#skip-navbar_bottom" title="Skip navigation links"></A>
<TABLE BORDER="0" WIDTH="100%" CELLPADDING="1" CELLSPACING="0" SUMMARY="">
<TR>
<TD COLSPAN=2 BGCOLOR="#EEEEFF" CLASS="NavBarCell1">
<A NAME="navbar_bottom_firstrow"><!-- --></A>
<TABLE BORDER="0" CELLPADDING="0" CELLSPACING="3" SUMMARY="">
  <TR ALIGN="center" VALIGN="top">
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../overview-summary.html"><FONT CLASS="NavBarFont1"><B>Overview</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-summary.html"><FONT CLASS="NavBarFont1"><B>Package</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../edu/berkeley/compbio/sequtils/NotEnoughSequenceException.html" title="class in edu.berkeley.compbio.sequtils"><FONT CLASS="NavBarFont1"><B>Class</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#FFFFFF" CLASS="NavBarCell1Rev"> &nbsp;<FONT CLASS="NavBarFont1Rev"><B>Use</B></FONT>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../package-tree.html"><FONT CLASS="NavBarFont1"><B>Tree</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../deprecated-list.html"><FONT CLASS="NavBarFont1"><B>Deprecated</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../index-all.html"><FONT CLASS="NavBarFont1"><B>Index</B></FONT></A>&nbsp;</TD>
  <TD BGCOLOR="#EEEEFF" CLASS="NavBarCell1">    <A HREF="../../../../../help-doc.html"><FONT CLASS="NavBarFont1"><B>Help</B></FONT></A>&nbsp;</TD>
  </TR>
</TABLE>
</TD>
<TD ALIGN="right" VALIGN="top" ROWSPAN=3><EM>
</EM>
</TD>
</TR>

<TR>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
&nbsp;PREV&nbsp;
&nbsp;NEXT</FONT></TD>
<TD BGCOLOR="white" CLASS="NavBarCell2"><FONT SIZE="-2">
  <A HREF="../../../../../index.html?edu/berkeley/compbio/sequtils//class-useNotEnoughSequenceException.html" target="_top"><B>FRAMES</B></A>  &nbsp;
&nbsp;<A HREF="NotEnoughSequenceException.html" target="_top"><B>NO FRAMES</B></A>  &nbsp;
&nbsp;<SCRIPT type="text/javascript">
  <!--
  if(window==top) {
    document.writeln('<A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>');
  }
  //-->
</SCRIPT>
<NOSCRIPT>
  <A HREF="../../../../../allclasses-noframe.html"><B>All Classes</B></A>
</NOSCRIPT>


</FONT></TD>
</TR>
</TABLE>
<A NAME="skip-navbar_bottom"></A>
<!-- ======== END OF BOTTOM NAVBAR ======= -->

<HR>
Copyright &#169; 2006&#x2013;2014 <a href="http://www.davidsoergel.com">David Soergel</a>. All rights reserved.
</BODY>
</HTML>
